<!DOCTYPE html>
<html lang="en"><head><meta charset="UTF-8"/><meta name="viewport" content="width=device-width, initial-scale=1.0"/><title>Home · JHistint Documentation</title><script data-outdated-warner src="assets/warner.js"></script><link href="https://cdnjs.cloudflare.com/ajax/libs/lato-font/3.0.0/css/lato-font.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/juliamono/0.045/juliamono.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.4/css/fontawesome.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.4/css/solid.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.4/css/brands.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.24/katex.min.css" rel="stylesheet" type="text/css"/><script>documenterBaseURL="."</script><script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" data-main="assets/documenter.js"></script><script src="siteinfo.js"></script><script src="../versions.js"></script><link class="docs-theme-link" rel="stylesheet" type="text/css" href="assets/themes/documenter-dark.css" data-theme-name="documenter-dark" data-theme-primary-dark/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="assets/themes/documenter-light.css" data-theme-name="documenter-light" data-theme-primary/><script src="assets/themeswap.js"></script></head><body><div id="documenter"><nav class="docs-sidebar"><div class="docs-package-name"><span class="docs-autofit"><a href>JHistint Documentation</a></span></div><form class="docs-search" action="search/"><input class="docs-search-query" id="documenter-search-query" name="q" type="text" placeholder="Search docs"/></form><ul class="docs-menu"><li class="is-active"><a class="tocitem" href>Home</a><ul class="internal"><li><a class="tocitem" href="#Package-Structure"><span>Package Structure</span></a></li><li><a class="tocitem" href="#JHistint-Collections"><span>JHistint Collections</span></a></li><li><a class="tocitem" href="#Package-Installation"><span>Package Installation</span></a></li><li><a class="tocitem" href="#Package-Installation-Julia-Registries-(In-Progress)"><span>Package Installation Julia Registries (In Progress)</span></a></li><li><a class="tocitem" href="#Download-Slides-Main-Functions-(JHistint.jl)"><span>Download Slides Main Functions (JHistint.jl)</span></a></li><li><a class="tocitem" href="#Cell-Segmentation-Slides-Main-Functions-(JHistint.jl)"><span>Cell Segmentation Slides Main Functions (JHistint.jl)</span></a></li><li><a class="tocitem" href="#SOPHYSM-Main-Function-(JHistint.jl)"><span>SOPHYSM Main Function (JHistint.jl)</span></a></li><li><a class="tocitem" href="#Support-Functions-for-Cell-Segmentation-(segmentationManager.jl)"><span>Support Functions for Cell Segmentation (segmentationManager.jl)</span></a></li><li><a class="tocitem" href="#SOPHYSM-Support-Function-for-Cell-Segmentation-(segmentationManager.jl)"><span>SOPHYSM Support Function for Cell Segmentation (segmentationManager.jl)</span></a></li><li><a class="tocitem" href="#Support-Functions-for-Graph-(graphManager.jl)"><span>Support Functions for Graph (graphManager.jl)</span></a></li><li><a class="tocitem" href="#Support-Functions-for-Tessellation-(tessellationManager.jl)"><span>Support Functions for Tessellation (tessellationManager.jl)</span></a></li><li><a class="tocitem" href="#Support-Functions-for-Noise-Reduction-(noiseManager.jl)"><span>Support Functions for Noise Reduction (noiseManager.jl)</span></a></li><li><a class="tocitem" href="#Support-Functions-for-DataBase-(dbManager.jl)"><span>Support Functions for DataBase (dbManager.jl)</span></a></li><li><a class="tocitem" href="#SOPHYSM-Support-Functions-for-DataBase-(dbManager.jl)"><span>SOPHYSM Support Functions for DataBase (dbManager.jl)</span></a></li><li><a class="tocitem" href="#API-Support-Functions-(apiManager.jl)"><span>API Support Functions (apiManager.jl)</span></a></li><li><a class="tocitem" href="#ZIP-Support-Functions-(zipManager.jl)"><span>ZIP Support Functions (zipManager.jl)</span></a></li></ul></li></ul><div class="docs-version-selector field has-addons"><div class="control"><span class="docs-label button is-static is-size-7">Version</span></div><div class="docs-selector control is-expanded"><div class="select is-fullwidth is-size-7"><select id="documenter-version-selector"></select></div></div></div></nav><div class="docs-main"><header class="docs-navbar"><nav class="breadcrumb"><ul class="is-hidden-mobile"><li class="is-active"><a href>Home</a></li></ul><ul class="is-hidden-tablet"><li class="is-active"><a href>Home</a></li></ul></nav><div class="docs-right"><a class="docs-edit-link" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/main/docs_builder/src/index.md" title="Edit on GitHub"><span class="docs-icon fab"></span><span class="docs-label is-hidden-touch">Edit on GitHub</span></a><a class="docs-settings-button fas fa-cog" id="documenter-settings-button" href="#" title="Settings"></a><a class="docs-sidebar-button fa fa-bars is-hidden-desktop" id="documenter-sidebar-button" href="#"></a></div></header><article class="content" id="documenter-page"><h1 id="JHistint.jl-Julia-Histopathology-Interface"><a class="docs-heading-anchor" href="#JHistint.jl-Julia-Histopathology-Interface">JHistint.jl - Julia Histopathology Interface</a><a id="JHistint.jl-Julia-Histopathology-Interface-1"></a><a class="docs-heading-anchor-permalink" href="#JHistint.jl-Julia-Histopathology-Interface" title="Permalink"></a></h1><p>Julia interface for implementing the REST APIs available on the Cancer Slide Digital Archive (CDSA) portal for downloading histological images available in The Cancer Genome Atlas (TCGA). The Cancer Slide Digital Archive (CDSA) is a web platform for support, sharing, and analysis of digital pathological data. Currently, it hosts over 23,000 images associated with the data available on &quot;The Cancer Genome Atlas&quot; Data Portal. The library includes functions for managing image-processing algorithms for cellular and nuclei segmentation, constructing graph and the corresponding adjacency matrix, building tessellation and interfacing with J-Space.jl package to simulate the spatial growth and the genomic evolution of a cell population and the experiment of sequencing the genome of the sampled cells.</p><p>Link GitHub repository: <a href="https://github.com/niccolo99mandelli/JHistint.jl">JHistint.jl</a></p><p>Link GitHub repository, avaiable on <code>spatial-input</code> branch: <a href="https://github.com/niccolo99mandelli/J-Space.jl">J-Space.jl</a></p><p>CDSA Portal: <a href="https://api.digitalslidearchive.org/#collections">Click Here</a>    </p><p>Repository containing the data mapped in the portal: <a href="https://cancer.digitalslidearchive.org/#!/CDSA/acc/TCGA-OR-A5J1">Click Here</a></p><p>Guide to using the APIs: <a href="https://api.digitalslidearchive.org/api/v1">Click Here</a></p><h2 id="Package-Structure"><a class="docs-heading-anchor" href="#Package-Structure">Package Structure</a><a id="Package-Structure-1"></a><a class="docs-heading-anchor-permalink" href="#Package-Structure" title="Permalink"></a></h2><ul><li>The <code>case</code> and <code>collection</code> folders store metadata in <code>.json</code> format for individual cases and collections available on the TCGA Data Portal. The <code>collection</code> folder is structured as follows:<ul><li><code>collectionlist.json</code> = Stores access data (metadata) for collections (Projects in TCGA).</li><li><code>collection_name.json</code> = Stores access data (metadata) for a single collection. The <code>.json</code> file is generated based on the collection chosen by the user.</li></ul></li><li>The <code>case</code> folder is structured as follows:<ul><li><code>collection_name.json</code> = Stores all metadata related to cases associated with the collection selected by the user.</li></ul></li><li>The <code>slides</code> folder stores histological images related to individual cases. The images are organized based on collection (<code>TCGA-chol</code>, <code>TCGA-esca</code>, etc.), and the individual case being analyzed (<code>TCGA-2H-A9GF</code>, <code>TCGA-2H-A9GG</code>, etc.). Within each folder related to the case, the slides are stored in compressed <code>.zip</code> files. The format of each individual slide is <code>.tif</code>. The folder names related to the cases correspond to the values of the <code>Case ID</code> field listed in the TCGA Data Portal. The names of the <code>.zip</code> files located in each folder refer to the <code>Sample ID</code> attribute associated with the patient. The slide name is given by concatenating the <code>Slide ID</code> and <code>Slide UUID</code> attributes that can be found in the lower section of the web page dedicated to the generic case <code>TCGA-XX-YYYY</code>.</li></ul><pre><code class="nohighlight hljs">Example: TCGA-02-0001-01C-01-TS1.zip  
  - 02 = refers to the TSS (Tissue Source Site).  
  - 0001 = refers to the code associated with the Participant, an alphanumeric string.  
  - 01 = refers to the Sample Type. The values associated with tumor samples are in the range 01-09. 10-19 indicates the range for non-diseased normal samples. 20-29 indicates samples currently under control.  
  - C = refers to the Vial field related to the ordering of the sample in the sample sequence. Values range from A-Z.  
  - 01 = refers to the Portion field related to the ordering of the analyzed portions associated with a sample. It takes values in the range 01-99.  
  - TS1 = refers to the Slide field related to the type of image. The values that can be assumed are TS (Top Slide), BS (Bottom Slide), and MS (Middle Slide). The alphanumeric value indicates the slide ordering.</code></pre><h2 id="JHistint-Collections"><a class="docs-heading-anchor" href="#JHistint-Collections">JHistint Collections</a><a id="JHistint-Collections-1"></a><a class="docs-heading-anchor-permalink" href="#JHistint-Collections" title="Permalink"></a></h2><p>The available collections are:</p><ul><li>TCGA-BRCA = Breast Invasive Carcinoma (Breast)</li><li>TCGA-OV = Ovarian Serous Cystadenocarcinoma (Ovary)</li><li>TCGA-LUAD = Lung Adenocarcinoma (Bronchus and Lung)</li><li>TCGA-UCEC = Uterine Corpus Endometrial Carcinoma (Corpus uteri)</li><li>TCGA-GBM = Glioblastoma Multiforme (Brain)</li><li>TCGA-HSNC = Head and Neck Squamous Cell Carcinoma (Larynx, Lip, Tonsil, Gum, Other and unspecified parths of mouth)</li><li>TCGA-KIRC = Kidney Renal Clear Cell Carcinoma (Kidney)</li><li>TCGA-LGG = Brain Lower Grade Glioma (Brain)</li><li>TCGA-LUSC = Lung Squamous Cell Carcinoma (Bronchus and lung)</li><li>TCGA-TCHA = Thyroid Carcinoma (Thyroid gland)</li><li>TCGA-PRAD = Prostate Adenocarcinoma (Prostate gland)</li><li>TCGA-SKCM = Skin Cutaneous Melanoma (Skin)</li><li>TCGA-COAD = Colon Adenocarcinoma (Colon)</li><li>TCGA-STAD = Stomach Adenocarcinoma (Stomach)</li><li>TCGA-BLCA = Bladder Urothelial Carcinoma (Bladder)</li><li>TCGA-LIHC = Liver Hepatocellular Carcinoma (Liver and intrahepatic bile ducts)  </li><li>TCGA-CESC = Cervical Squamous Cell Carcinoma and Endocervical Adenocarcinoma (Cervix uteri)</li><li>TCGA-KIRP = Kidney Renal Papillary Cell Carcinoma (Kidney)</li><li>TCGA-SARC = Sarcoma (Various)</li><li>TCGA-ESCA = Esophageal Carcinoma (Esophagus)</li><li>TCGA-PAAD = Pancreatic Adenocarcinoma (Pancreas)</li><li>TCGA-READ = Rectum Adenocarcinoma (Rectum)</li><li>TCGA-PCPG = Pheochromocytoma and Paraganglioma (Adrenal gland)</li><li>TCGA-TGCT = Testicular Germ Cell Tumors (Testis)</li><li>TCGA-THYM = Thymoma (Thymus)</li><li>TCGA-ACC = Adrenocortical Carcinoma -Adenomas and Adenocarcinomas (Adrenal gland)</li><li>TCGA-MESO = Mesothelioma (Heart, mediastinum and pleura)</li><li>TCGA-UVM = Uveal Melanoma (Eye and adnexa)</li><li>TCGA-KICH = Kidney Chromophobe (Kidney)</li><li>TCGA-UCS = Uterine Carcinosarcoma (Uterus, NOS)</li><li>TCGA-CHOL = Cholangiocarcinoma (Liver and intrahepatic bile ducts, Other and unspecified part of biliary track)</li><li>TCGA-DLBC = Lymphoid Neoplasm Diffuse Large B-cell Lymphoma (Various)</li></ul><p>To download a specific collection, just indicate the name of the collection: <code>BRCA</code>, <code>OV</code>, <code>LUAD</code>.</p><h2 id="Package-Installation"><a class="docs-heading-anchor" href="#Package-Installation">Package Installation</a><a id="Package-Installation-1"></a><a class="docs-heading-anchor-permalink" href="#Package-Installation" title="Permalink"></a></h2><ul><li>Step 1 - Install <code>J-Space</code> from <code>spatial-input</code> branch:</li></ul><pre><code class="nohighlight hljs">(@v1.8) pkg &gt; add https://github.com/BIMIB-DISCo/J-Space.jl.git#spatial-input</code></pre><ul><li>Step 2 - Install <code>JHistint</code> from GitHub Repository:</li></ul><pre><code class="nohighlight hljs">(@v1.8) pkg &gt; add https://github.com/niccolo99mandelli/JHistint.jl.git
julia &gt; using JHistint</code></pre><h2 id="Package-Installation-Julia-Registries-(In-Progress)"><a class="docs-heading-anchor" href="#Package-Installation-Julia-Registries-(In-Progress)">Package Installation Julia Registries (In Progress)</a><a id="Package-Installation-Julia-Registries-(In-Progress)-1"></a><a class="docs-heading-anchor-permalink" href="#Package-Installation-Julia-Registries-(In-Progress)" title="Permalink"></a></h2><p>The <code>JHistint</code> package is available in the Julia Registries and can be installed as follows:</p><pre><code class="nohighlight hljs">julia &gt; using Pkg
julia &gt; Pkg.add(&quot;JHistint&quot;)
julia &gt; using JHistint</code></pre><p>Otherwise, type <code>]</code> in the Julia REPL and execute:</p><pre><code class="nohighlight hljs">(@v1.8) pkg &gt; add JHistint
julia &gt; using JHistint</code></pre><h2 id="Download-Slides-Main-Functions-(JHistint.jl)"><a class="docs-heading-anchor" href="#Download-Slides-Main-Functions-(JHistint.jl)">Download Slides Main Functions (JHistint.jl)</a><a id="Download-Slides-Main-Functions-(JHistint.jl)-1"></a><a class="docs-heading-anchor-permalink" href="#Download-Slides-Main-Functions-(JHistint.jl)" title="Permalink"></a></h2><article class="docstring"><header><a class="docstring-binding" id="JHistint.download_single_collection-Tuple{AbstractString}" href="#JHistint.download_single_collection-Tuple{AbstractString}"><code>JHistint.download_single_collection</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">download_single_collection(collection_name::AbstractString)</code></pre><p>Function for downloading histological slides associated with a collection available in TCGA.</p><p><strong>Arguments</strong></p><ul><li><code>collection_name::AbstractString</code> = Collection of TCGA data to download</li></ul><p>the histological slides.</p><p><strong>Notes</strong></p><p>The function evaluates the <code>collection_name</code> argument, and in case of an invalid collection, considers the configuration in the <code>Config.toml</code> file. The value set in the package is <code>default</code>.</p><pre><code class="language-julia hljs"># Examples with valid input
julia&gt; JHistint.download_single_collection(&quot;acc&quot;)
julia&gt; JHistint.download_single_collection(&quot;bLca&quot;)</code></pre><pre><code class="language-julia hljs"># Examples with invalid input
julia&gt; JHistint.download_single_collection(&quot;ac&quot;)
julia&gt; JHistint.download_single_collection(&quot;&quot;)</code></pre></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/JHistint.jl#L54-L78">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="JHistint.download_all_collection-Tuple{}" href="#JHistint.download_all_collection-Tuple{}"><code>JHistint.download_all_collection</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">download_all_collection()</code></pre><p>Function for downloading histological slides associated with all collections available in TCGA.</p><pre><code class="language-julia hljs"># Examples with valid input
julia&gt; JHistint.download_all_collection()</code></pre></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/JHistint.jl#L245-L255">source</a></section></article><h3 id="Download-Slides-SOPHYSM-Functions-(JHistint.jl)"><a class="docs-heading-anchor" href="#Download-Slides-SOPHYSM-Functions-(JHistint.jl)">Download Slides SOPHYSM Functions (JHistint.jl)</a><a id="Download-Slides-SOPHYSM-Functions-(JHistint.jl)-1"></a><a class="docs-heading-anchor-permalink" href="#Download-Slides-SOPHYSM-Functions-(JHistint.jl)" title="Permalink"></a></h3><article class="docstring"><header><a class="docstring-binding" id="JHistint.download_single_collection_SOPHYSM-Tuple{AbstractString, AbstractString}" href="#JHistint.download_single_collection_SOPHYSM-Tuple{AbstractString, AbstractString}"><code>JHistint.download_single_collection_SOPHYSM</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">download_single_collection_SOPHYSM(collection_name::AbstractString, path_to_save::AbstractString)</code></pre><p>Function for downloading histological slides in SOPYHSM_app associated with a collection available in TCGA.</p><p><strong>Arguments</strong></p><ul><li><code>collection_name::AbstractString</code> = Collection of TCGA data to download the</li></ul><p>histological slides.</p><ul><li><code>path_to_save::AbstractString</code> = Local folder path for saving</li></ul><p>histological slides.</p><p><strong>Notes</strong></p><p>The function evaluates the <code>collection_name</code> argument, and in case of an invalid collection, considers the configuration in the <code>Config.toml</code> file. The value set in the package is <code>default</code>.</p><pre><code class="language-julia hljs"># Examples with valid input
julia&gt; JHistint.download_single_collection_SOPHYSM(&quot;acc&quot;, &quot;C:\...&quot;)
julia&gt; JHistint.download_single_collection_SOPHYSM(&quot;bLca&quot;, &quot;C:\...&quot;)</code></pre><pre><code class="language-julia hljs"># Examples with invalid input
julia&gt; JHistint.download_single_collection_SOPHYSM(&quot;ac&quot;, &quot;C:\...&quot;)
julia&gt; JHistint.download_single_collection_SOPHYSM(&quot;&quot;, &quot;C:\...&quot;)</code></pre></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/JHistint.jl#L171-L197">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="JHistint.download_all_collection_SOPHYSM-Tuple{AbstractString}" href="#JHistint.download_all_collection_SOPHYSM-Tuple{AbstractString}"><code>JHistint.download_all_collection_SOPHYSM</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">download_all_collection_SOPHYSM(path_to_save::AbstractString)</code></pre><p>Function for downloading histological slides associated with all collections available in TCGA.</p><p><strong>Arguments</strong></p><ul><li><code>path_to_save::AbstractString</code> = Local folder path for saving</li></ul><p>histological slides.</p><pre><code class="language-julia hljs"># Examples with valid input
julia&gt; JHistint.download_all_collection_SOPHYSM(&quot;C:\...&quot;)</code></pre></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/JHistint.jl#L299-L313">source</a></section></article><h2 id="Cell-Segmentation-Slides-Main-Functions-(JHistint.jl)"><a class="docs-heading-anchor" href="#Cell-Segmentation-Slides-Main-Functions-(JHistint.jl)">Cell Segmentation Slides Main Functions (JHistint.jl)</a><a id="Cell-Segmentation-Slides-Main-Functions-(JHistint.jl)-1"></a><a class="docs-heading-anchor-permalink" href="#Cell-Segmentation-Slides-Main-Functions-(JHistint.jl)" title="Permalink"></a></h2><article class="docstring"><header><a class="docstring-binding" id="JHistint.slide_cell_segmentation_without_download-Tuple{AbstractString}" href="#JHistint.slide_cell_segmentation_without_download-Tuple{AbstractString}"><code>JHistint.slide_cell_segmentation_without_download</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">slide_cell_segmentation_without_download(collection_name::AbstractString)</code></pre><p>Function for performing cell segmentation on histopathological slides present in the <code>JHistint_DB</code> database associated with the collection name provided as an argument. After generating the segmented slide, the function proceeds with constructing and saving the corresponding graph and adjacency matrix.</p><p><strong>Arguments</strong></p><ul><li><code>collection_name::AbstractString</code> = Collection of TCGA data to download</li></ul><p>the histological slides.</p><p><strong>Notes</strong></p><p>The function utilizes the <code>JHistint_DB</code> database for performing cell segmentation on the histopathological slides associated with the provided collection name. It generates a segmented slide and constructs a corresponding graph and adjacency matrix. The output files are saved in a user-defined directory. The function may take a considerable amount of time to complete, depending on the size of the slides and the complexity of the segmentation algorithm. For each slide in the database, cell segmentation is performed using the <code>apply_segmentation_without_download</code> function, and the path where the result is saved is stored in the database using the <code>load_seg_slide</code> function. The segmentation process is defined in 4 steps:</p><ul><li>LOAD SLIDE ... (slide_id)</li><li>APPLY SEGMENTATION ... (slide_id)</li><li>BUILD GRAPH ... (slide_id)</li><li>BUILD &amp; SAVE ADJACENCY MATRIX ... (slide_id)</li><li>J-SPACE features ... (slide_id)</li></ul><p>The adjacency matrix is saved in the same directory as the original image in text format. Finally, a confirmation message is printed for each segmented slide. Unlike the <code>slide_cell_segmentation_with_download</code> function, this function does not involve the creation and download of the segmented image.</p><pre><code class="language-julia hljs"># Examples with valid input
julia&gt; JHistint.slide_cell_segmentation_without_download(&quot;acc&quot;)
julia&gt; JHistint.slide_cell_segmentation_without_download(&quot;bLca&quot;)</code></pre><pre><code class="language-julia hljs"># Examples with invalid input
julia&gt; JHistint.slide_cell_segmentation_without_download(&quot;ac&quot;)
julia&gt; JHistint.slide_cell_segmentation_without_download(&quot;&quot;)</code></pre></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/JHistint.jl#L358-L401">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="JHistint.slide_cell_segmentation_with_download-Tuple{AbstractString}" href="#JHistint.slide_cell_segmentation_with_download-Tuple{AbstractString}"><code>JHistint.slide_cell_segmentation_with_download</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">slide_cell_segmentation_with_download(collection_name::AbstractString)</code></pre><p>Function for performing cell segmentation on histopathological slides present in the <code>JHistint_DB</code> database associated with the collection name provided as an argument. The function downloads the segmented slide, which is placed in the same directory as the original slide. After generating the segmented slide, the function proceeds with constructing and saving the corresponding graph and adjacency matrix.</p><p><strong>Arguments</strong></p><ul><li><code>collection_name::AbstractString</code> = TCGA data collection for which to</li></ul><p>perform cell segmentation.</p><p><strong>Notes</strong></p><p>The function utilizes the <code>JHistint_DB</code> database for performing cell segmentation on the histopathological slides associated with the provided collection name. It generates a segmented slide and constructs a corresponding graph and adjacency matrix. The output files are saved in a user-defined directory. The function may take a considerable amount of time to complete, depending on the size of the slides and the complexity of the segmentation algorithm. For each slide in the database, cell segmentation is performed using the <code>apply_segmentation_with_download</code> function, and the path where the result is saved is stored in the database using the <code>load_seg_slide</code> function. The segmentation process is similar to that described in the <code>slide_cell_segmentation_without_download</code> function, with the added step of downloading the segmented image and placing it in the same directory as the original slide. The segmentation process is defined in 6 steps:</p><ul><li>LOAD SLIDE ... (slide_id)</li><li>APPLY SEGMENTATION ... (slide_id)</li><li>BUILD SEGMENTED SLIDE ... (slide_id)</li><li>BUILD GRAPH ... (slide_id)</li><li>BUILD &amp; SAVE ADJACENCY MATRIX ... (slide_id)</li><li>SAVE SEGMENTED SLIDE ... (slide_id)</li><li>J-SPACE features ... (slide_id)</li></ul><p>The adjacency matrix is saved in text format in the same directory as both the original and segmented images. Finally, a confirmation message is printed for each segmented slide.</p><pre><code class="language-julia hljs"># Examples with valid input
julia&gt; JHistint.slide_cell_segmentation_with_download(&quot;acc&quot;)
julia&gt; JHistint.slide_cell_segmentation_with_download(&quot;bLca&quot;)</code></pre><pre><code class="language-julia hljs"># Examples with invalid input
julia&gt; JHistint.slide_cell_segmentation_with_download(&quot;ac&quot;)
julia&gt; JHistint.slide_cell_segmentation_with_download(&quot;&quot;)</code></pre></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/JHistint.jl#L452-L499">source</a></section></article><h2 id="SOPHYSM-Main-Function-(JHistint.jl)"><a class="docs-heading-anchor" href="#SOPHYSM-Main-Function-(JHistint.jl)">SOPHYSM Main Function (JHistint.jl)</a><a id="SOPHYSM-Main-Function-(JHistint.jl)-1"></a><a class="docs-heading-anchor-permalink" href="#SOPHYSM-Main-Function-(JHistint.jl)" title="Permalink"></a></h2><article class="docstring"><header><a class="docstring-binding" id="JHistint.start_segmentation_SOPHYSM_tessellation-Tuple{AbstractString, AbstractString, Float64, Float64, Float32, Float32}" href="#JHistint.start_segmentation_SOPHYSM_tessellation-Tuple{AbstractString, AbstractString, Float64, Float64, Float32, Float32}"><code>JHistint.start_segmentation_SOPHYSM_tessellation</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">start_segmentation_SOPHYSM_tessellation(filepath_input::AbstractString,
                                filepath_output::AbstractString,
                                thresholdGray::Float64,
                                thresholdMarker::Float64,
                                min_threshold::Float32,
                                max_threshold::Float32)</code></pre><p>Initiates the SOPHYSM segmentation process for histological image using tessellation process.</p><p><strong>Arguments</strong></p><ul><li><code>filepath_input::AbstractString</code>: The file path to the input histological</li></ul><p>image to be segmented.</p><ul><li><code>filepath_output::AbstractString</code>: The file path where the segmented results</li></ul><p>and related data will be saved.</p><ul><li><code>thresholdGray::Float64</code>: The grayscale threshold used for initial</li></ul><p>image processing.</p><ul><li><code>thresholdMarker::Float64</code>: The marker threshold for identifying</li></ul><p>cellular structures.</p><ul><li><code>min_threshold</code>: Minimal threshold for considering segments area.</li><li><code>max_threshold</code>: Maximal threshold for considering segments area.</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/JHistint.jl#L550-L572">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="JHistint.start_segmentation_SOPHYSM_graph-Tuple{AbstractString, AbstractString, Float64, Float64, Float32, Float32}" href="#JHistint.start_segmentation_SOPHYSM_graph-Tuple{AbstractString, AbstractString, Float64, Float64, Float32, Float32}"><code>JHistint.start_segmentation_SOPHYSM_graph</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">start_segmentation_SOPHYSM_graph(filepath_input::AbstractString,
                                filepath_output::AbstractString,
                                thresholdGray::Float64,
                                thresholdMarker::Float64,
                                min_threshold::Float32,
                                max_threshold::Float32)</code></pre><p>Initiates the SOPHYSM segmentation process for histological image using graph construction.</p><p><strong>Arguments</strong></p><ul><li><code>filepath_input::AbstractString</code>: The file path to the input histological</li></ul><p>image to be segmented.</p><ul><li><code>filepath_output::AbstractString</code>: The file path where the segmented results</li></ul><p>and related data will be saved.</p><ul><li><code>thresholdGray::Float64</code>: The grayscale threshold used for initial</li></ul><p>image processing.</p><ul><li><code>thresholdMarker::Float64</code>: The marker threshold for identifying</li></ul><p>cellular structures.</p><ul><li><code>min_threshold</code>: Minimal threshold for considering segments area.</li><li><code>max_threshold</code>: Maximal threshold for considering segments area.</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/JHistint.jl#L587-L609">source</a></section></article><h2 id="Support-Functions-for-Cell-Segmentation-(segmentationManager.jl)"><a class="docs-heading-anchor" href="#Support-Functions-for-Cell-Segmentation-(segmentationManager.jl)">Support Functions for Cell Segmentation (segmentationManager.jl)</a><a id="Support-Functions-for-Cell-Segmentation-(segmentationManager.jl)-1"></a><a class="docs-heading-anchor-permalink" href="#Support-Functions-for-Cell-Segmentation-(segmentationManager.jl)" title="Permalink"></a></h2><article class="docstring"><header><a class="docstring-binding" id="JHistint.apply_segmentation_without_download-Tuple{Tuple{String, Vector{UInt8}, String}}" href="#JHistint.apply_segmentation_without_download-Tuple{Tuple{String, Vector{UInt8}, String}}"><code>JHistint.apply_segmentation_without_download</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">apply_segmentation_without_download(slide_info::Tuple{String, Vector{UInt8}, String})</code></pre><p>The function performs the segmentation of a histological image, generates its corresponding graph, and translates it into a symmetric adjacency matrix with only 0s and 1s. Define, also, the dataframe associated with labels and edges.</p><p><strong>Arguments</strong></p><ul><li><code>slide_info::Tuple{String, Vector{UInt8}, String}</code>: A tuple containing the</li></ul><p>slide ID, the image obtained from the DB, and the path of the original image file.</p><p><strong>Return values</strong></p><ul><li><code>filepath_matrix</code>: The path where the adjacency matrix is stored in <code>.txt</code> format.</li><li><code>matrix</code>: The adjacency matrix constructed from the segmentation.</li></ul><p><strong>Notes</strong></p><p>The function uses the watershed segmentation algorithm to segment the image into different groups of pixels. Segmentation is performed using a feature transformation of the image (<code>feature_transform</code>) and labeling of connected components. The distance between the different regions is then calculated, and an adjacency graph of the regions is constructed using the <code>region_adjacency_graph</code> function. The resulting graph is then transformed into an Int adjacency matrix using the <code>weighted_graph_to_adjacency_matrix</code> function and saved to the path of the original image.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/segmentationManager.jl#L30-L55">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="JHistint.apply_segmentation_with_download-Tuple{Tuple{String, Vector{UInt8}, String}}" href="#JHistint.apply_segmentation_with_download-Tuple{Tuple{String, Vector{UInt8}, String}}"><code>JHistint.apply_segmentation_with_download</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">apply_segmentation_with_download(slide_info::Tuple{String, Vector{UInt8}, String})</code></pre><p>The function performs segmentation of a histological image, saves the segmented image in <code>.png</code> format, generates the corresponding graph, and translates it into an adjacency matrix. Define, also, the dataframe associated with labels and edges.</p><p><strong>Arguments</strong></p><ul><li><code>slide_info::Tuple{String, Vector{UInt8}, String}</code>: Tuple containing the</li></ul><p>slide ID, the image itself obtained from the DB, and the path of the original image file.</p><p><strong>Return values</strong></p><ul><li><code>filepath_seg</code>: The path where the segmented image is stored in <code>.tif</code> format.</li><li><code>filepath_matrix</code>: The path where the graph is stored in <code>.txt</code> format.</li><li><code>matrix</code>: The adjacency matrix constructed from the segmentation.</li></ul><p><strong>Notes</strong></p><p>The function uses the watershed segmentation algorithm to segment the image into different groups of pixels. Segmentation is performed using an image feature transformation (<code>feature_transform</code>) and connected component labeling. The distance between different regions is then calculated, and an adjacency graph of the regions is constructed using the <code>region_adjacency_graph</code> function. The obtained graph is transformed into an adjacency matrix using the <code>weighted_graph_to_adjacency_matrix</code> function, which is saved in the path of the original image. Finally, a segmented <code>.png</code> image is saved, and the path of the segmented slide file is returned. It performs also the visualizations of the graph with vertices and edges saved as &quot;<em>graph</em>vertex.png&quot; and &quot;<em>graph</em>edges.png&quot; in the output directory.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/segmentationManager.jl#L85-L116">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="JHistint.get_random_color-Tuple{Any}" href="#JHistint.get_random_color-Tuple{Any}"><code>JHistint.get_random_color</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">get_random_color(seed)</code></pre><p>Function to return a random 8-bit RGB format color, using a specified seed.</p><p><strong>Arguments</strong></p><ul><li><code>seed</code>: An integer used to initialize the random number generator.</li></ul><p>If two calls to the function use the same seed, the same color will be generated.</p><p><strong>Return value</strong></p><p>The function returns a random 8-bit RGB format color.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/segmentationManager.jl#L13-L24">source</a></section></article><h2 id="SOPHYSM-Support-Function-for-Cell-Segmentation-(segmentationManager.jl)"><a class="docs-heading-anchor" href="#SOPHYSM-Support-Function-for-Cell-Segmentation-(segmentationManager.jl)">SOPHYSM Support Function for Cell Segmentation (segmentationManager.jl)</a><a id="SOPHYSM-Support-Function-for-Cell-Segmentation-(segmentationManager.jl)-1"></a><a class="docs-heading-anchor-permalink" href="#SOPHYSM-Support-Function-for-Cell-Segmentation-(segmentationManager.jl)" title="Permalink"></a></h2><article class="docstring"><header><a class="docstring-binding" id="JHistint.apply_segmentation_SOPHYSM_tessellation-Tuple{AbstractString, AbstractString, Float64, Float64, Float32, Float32}" href="#JHistint.apply_segmentation_SOPHYSM_tessellation-Tuple{AbstractString, AbstractString, Float64, Float64, Float32, Float32}"><code>JHistint.apply_segmentation_SOPHYSM_tessellation</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">apply_segmentation_SOPHYSM_tessellation(filepath_input::AbstractString,
                                filepath_output::AbstractString,
                                thresholdGray::Float64,
                                thresholdMarker::Float64,
                                min_threshold::Float32,
                                max_threshold::Float32)</code></pre><p>The function performs segmentation of a histological slide, saves the segmented image in <code>.png</code> format, generates the corresponding tessellation, and translates it into an adjacency matrix and build the corresponding dataframe.</p><p><strong>Arguments</strong></p><ul><li><code>filepath_input::AbstractString</code>: The file path to the input image.</li><li><code>filepath_output::AbstractString</code>: The file path where the output files will be saved.</li><li><code>thresholdGray::Float64</code>: The grayscale threshold for image binarization.</li><li><code>thresholdMarker::Float64</code>: The threshold for marker-based segmentation.</li><li><code>min_threshold</code>: Minimal threshold for considering segments area.</li><li><code>max_threshold</code>: Maximal threshold for considering segments area.</li></ul><p><strong>Notes</strong></p><p>The function uses the watershed segmentation algorithm to segment the image into different groups of pixels. Segmentation is performed using an image feature transformation (<code>feature_transform</code>) and connected component labeling. The <code>apply_segmentation_SOPHYSM</code> function reads an <code>.tif</code> image, applies the SOPHYSM segmentation algorithm, and generates the following outputs:</p><ol><li>Segmented image saved as &quot;_seg.png&quot; in the output directory.</li><li>Dataframe containing label information saved as &quot;<em>dataframe</em>labels.csv&quot;</li></ol><p>in the output directory. Also, the dataframe contaning extra information about the segment and label computed by the segmentation algorithm.</p><ol><li>Adjacency matrix saved as &quot;.txt&quot; in the output directory.</li><li>Dataframe containing edge information saved as &quot;<em>dataframe</em>edges.csv&quot;</li></ol><p>in the output directory.</p><ol><li>Visualizations of the graph with vertices and edges saved as</li></ol><p>&quot;<em>graph</em>vertex.png&quot; and &quot;<em>graph</em>edges.png&quot; in the output directory.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/segmentationManager.jl#L183-L218">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="JHistint.apply_segmentation_SOPHYSM_graph-Tuple{AbstractString, AbstractString, Float64, Float64, Float32, Float32}" href="#JHistint.apply_segmentation_SOPHYSM_graph-Tuple{AbstractString, AbstractString, Float64, Float64, Float32, Float32}"><code>JHistint.apply_segmentation_SOPHYSM_graph</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">apply_segmentation_SOPHYSM_graph(filepath_input::AbstractString,
                                filepath_output::AbstractString,
                                thresholdGray::Float64,
                                thresholdMarker::Float64,
                                min_threshold::Float32,
                                max_threshold::Float32)</code></pre><p>The function performs segmentation of a histological slide, saves the segmented image in <code>.png</code> format, generates the corresponding graph, and translates it into an adjacency matrix and build the corresponding dataframe.</p><p><strong>Arguments</strong></p><ul><li><code>filepath_input::AbstractString</code>: The file path to the input image.</li><li><code>filepath_output::AbstractString</code>: The file path where the output files will be saved.</li><li><code>thresholdGray::Float64</code>: The grayscale threshold for image binarization.</li><li><code>thresholdMarker::Float64</code>: The threshold for marker-based segmentation.</li><li><code>min_threshold</code>: Minimal threshold for considering segments area.</li><li><code>max_threshold</code>: Maximal threshold for considering segments area.</li></ul><p><strong>Notes</strong></p><p>The function uses the watershed segmentation algorithm to segment the image into different groups of pixels. Segmentation is performed using an image feature transformation (<code>feature_transform</code>) and connected component labeling. The <code>apply_segmentation_SOPHYSM</code> function reads an <code>.tif</code> image, applies the SOPHYSM segmentation algorithm, and generates the following outputs:</p><ol><li>Segmented image saved as &quot;_seg.png&quot; in the output directory.</li><li>Dataframe containing label information saved as &quot;<em>dataframe</em>labels.csv&quot;</li></ol><p>in the output directory. Also, the dataframe contaning extra information about the segment and label computed by the segmentation algorithm.</p><ol><li>Adjacency matrix saved as &quot;.txt&quot; in the output directory.</li><li>Dataframe containing edge information saved as &quot;<em>dataframe</em>edges.csv&quot;</li></ol><p>in the output directory.</p><ol><li>Visualizations of the graph with vertices and edges saved as</li></ol><p>&quot;<em>graph</em>vertex.png&quot; and &quot;<em>graph</em>edges.png&quot; in the output directory.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/segmentationManager.jl#L308-L343">source</a></section></article><h2 id="Support-Functions-for-Graph-(graphManager.jl)"><a class="docs-heading-anchor" href="#Support-Functions-for-Graph-(graphManager.jl)">Support Functions for Graph (graphManager.jl)</a><a id="Support-Functions-for-Graph-(graphManager.jl)-1"></a><a class="docs-heading-anchor-permalink" href="#Support-Functions-for-Graph-(graphManager.jl)" title="Permalink"></a></h2><article class="docstring"><header><a class="docstring-binding" id="JHistint.region_adjacency_graph_JHistint-Tuple{SegmentedImage, Function, Float32, Float32}" href="#JHistint.region_adjacency_graph_JHistint-Tuple{SegmentedImage, Function, Float32, Float32}"><code>JHistint.region_adjacency_graph_JHistint</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">region_adjacency_graph_JHistint(s::SegmentedImage, weight_fn::Function,
                            min_threshold::Float32, max_threshold::Float32)</code></pre><p>Constructs a region adjacency graph (RAG) from the <code>SegmentedImage</code>. It returns the RAGalong with a Dict(label=&gt;vertex) map and a dataframe containing the information about the label. <code>weight_fn</code> is used to assign weights to the edges.</p><p><strong>Arguments :</strong></p><ul><li><code>s::SegmentedImage</code>: The input segmented image containing regions.</li><li><code>weight_fn::Function</code>: A function that calculates the weight between</li></ul><p>two adjacent regions. The function should accept two region labels as arguments and return a numeric value representing the weight.</p><p><strong>Return value :</strong></p><ul><li><code>G::SimpleWeightedGraph</code>: The adjacency graph between regions with weights</li></ul><p>on the edges.</p><ul><li><code>vert_map::Dict{Int, Int}</code>: A dictionary that maps region labels to nodes</li></ul><p>in the graph.</p><ul><li><code>df_label::DataFrame</code>: A DataFrame containing information about regions,</li></ul><p>including their identifiers, positions, colors, and areas.</p><p><strong>Notes:</strong></p><p>weight_fn(label1, label2): Returns a real number corresponding to the weight of the edge between label1 and label2.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/graphManager.jl#L37-L62">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="JHistint.weighted_graph_to_adjacency_matrix-Tuple{SimpleWeightedGraph{Int64, Float64}, Int64}" href="#JHistint.weighted_graph_to_adjacency_matrix-Tuple{SimpleWeightedGraph{Int64, Float64}, Int64}"><code>JHistint.weighted_graph_to_adjacency_matrix</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">weighted_graph_to_adjacency_matrix(G::SimpleWeightedGraph{Int64, Float64}, n::Int64)</code></pre><p>Converts a weighted graph represented as a <code>SimpleWeightedGraph</code> into an unweighted boolean adjacency matrix.</p><p><strong>Arguments:</strong></p><ul><li><code>G::SimpleWeightedGraph{Int64, Float64}</code>: Weighted graph represented as a</li></ul><p><code>SimpleWeightedGraph</code> with integer vertex labels and floating-point edge weights.</p><ul><li><code>n::Int64</code>: Number of nodes in the adjacency matrix.</li></ul><p><strong>Return value:</strong></p><ul><li><code>adjacency_matrix</code>: <code>Matrix{Int64}</code> boolean adjacency matrix.</li></ul><p><strong>Notes:</strong></p><p>The function returns an <code>n</code> x <code>n</code> adjacency matrix representing the unweighted graph. If nodes <code>i</code> and <code>j</code> are adjacent,the adjacency matrix will contain a value of <code>1</code> at position <code>(i,j)</code> and <code>(j,i)</code>. Otherwise, the adjacency matrix will contain a value of <code>0</code>.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/graphManager.jl#L200-L219">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="JHistint.weighted_graph_to_adjacency_matrix_weight-Tuple{SimpleWeightedGraph{Int64, Float64}, Int64}" href="#JHistint.weighted_graph_to_adjacency_matrix_weight-Tuple{SimpleWeightedGraph{Int64, Float64}, Int64}"><code>JHistint.weighted_graph_to_adjacency_matrix_weight</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">weighted_graph_to_adjacency_matrix_weight(G::SimpleWeightedGraph{Int64, Float64}, n::Int64)</code></pre><p>Converts a weighted graph represented as a <code>SimpleWeightedGraph</code> into an weighted adjacency matrix.</p><p><strong>Arguments:</strong></p><ul><li><code>G::SimpleWeightedGraph{Int64, Float64}</code>: Weighted graph represented as a</li></ul><p><code>SimpleWeightedGraph</code> with integer vertex labels and floating-point edge weights.</p><ul><li><code>n::Int64</code>: Number of nodes in the adjacency matrix.</li></ul><p><strong>Return value:</strong></p><ul><li><code>adjacency_matrix</code>: <code>Matrix{Float32}</code> boolean adjacency matrix.</li></ul><p><strong>Notes:</strong></p><p>The function returns an <code>n</code> x <code>n</code> adjacency matrix representing the weighted graph. If nodes <code>i</code> and <code>j</code> are adjacent,the adjacency matrix will contain a value associated with the edge weight at position <code>(i,j)</code> and <code>(j,i)</code>. Otherwise, the adjacency matrix will contain a value of <code>-1</code>.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/graphManager.jl#L233-L252">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="JHistint.build_dataframe_as_edgelist-Tuple{Matrix{Int64}, Vector{Int64}}" href="#JHistint.build_dataframe_as_edgelist-Tuple{Matrix{Int64}, Vector{Int64}}"><code>JHistint.build_dataframe_as_edgelist</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">build_dataframe_as_edgelist(mat::Matrix{Int64}, label_list::Vector{Int64})</code></pre><p>Builds a DataFrame representing an edge list from an input adjacency matrix <code>mat</code> and a list of labels <code>label_list</code>.</p><p><strong>Arguments:</strong></p><ul><li><code>mat::Matrix{Int64}</code>: The input adjacency matrix where elements represent</li></ul><p>connections between nodes.</p><ul><li><code>label_list::Vector{Int64}</code>: A list of node labels to consider when</li></ul><p>constructing the edge list.</p><p><strong>Return value:</strong></p><ul><li><code>df::DataFrame</code>: A DataFrame representing the edges in the graph,</li></ul><p>with columns &#39;origin&#39;, &#39;destination&#39;, and &#39;weight&#39; indicating the source node, target node, and edge weight, respectively.</p><p><strong>Notes:</strong></p><p>The <code>build_dataframe_as_edgelist</code> function constructs a DataFrame that represents the edges in a graph based on the adjacency matrix <code>mat</code>. It iterates through the upper triangular part of the matrix and adds edges to the DataFrame for non-zero values while considering only nodes with labels present in <code>label_list</code>.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/graphManager.jl#L274-L297">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="JHistint.build_dataframe_as_edgelist-Tuple{Matrix{Float32}, Vector{Int64}}" href="#JHistint.build_dataframe_as_edgelist-Tuple{Matrix{Float32}, Vector{Int64}}"><code>JHistint.build_dataframe_as_edgelist</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">build_dataframe_as_edgelist(mat::Matrix{Float32}, label_list::Vector{Int64})</code></pre><p>Builds a DataFrame representing an edge list from an input adjacency matrix <code>mat</code> and a list of labels <code>label_list</code>.</p><p><strong>Arguments:</strong></p><ul><li><code>mat::Matrix{Float32}</code>: The input adjacency matrix where elements represent</li></ul><p>connections between nodes. In this case, the matrix has Float value.</p><ul><li><code>label_list::Vector{Int64}</code>: A list of node labels to consider when</li></ul><p>constructing the edge list.</p><p><strong>Return value:</strong></p><ul><li><code>df::DataFrame</code>: A DataFrame representing the edges in the graph,</li></ul><p>with columns &#39;origin&#39;, &#39;destination&#39;, and &#39;weight&#39; indicating the source node, target node, and edge weight, respectively.</p><p><strong>Notes:</strong></p><p>The <code>build_dataframe_as_edgelist</code> function constructs a DataFrame that represents the edges in a graph based on the adjacency matrix <code>mat</code>. It iterates through the upper triangular part of the matrix and adds edges to the DataFrame for values different from <code>-1</code> while considering only nodes with labels present in <code>label_list</code>.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/graphManager.jl#L319-L342">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="JHistint.save_adjacency_matrix-Tuple{Matrix{Int64}, AbstractString}" href="#JHistint.save_adjacency_matrix-Tuple{Matrix{Int64}, AbstractString}"><code>JHistint.save_adjacency_matrix</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">save_adjacency_matrix(matrix::Matrix{Int64}, filepath_matrix::AbstractString)</code></pre><p>Function to save an adjacency matrix represented as an integer matrix to a text file.</p><p><strong>Arguments:</strong></p><ul><li><code>matrix::Matrix{Int64}</code>: The integer matrix representing the adjacency matrix.</li><li><code>filepath_matrix::AbstractString</code>: The file path represented as a string</li></ul><p>indicating where to save the matrix.</p><p><strong>Notes</strong></p><p>The function opens the file specified by the <code>filepath_matrix</code> path in write mode and writes the matrix in adjacency matrix format, where each row represents the adjacent nodes of a node. The numbers in the matrix are separated by spaces.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/graphManager.jl#L365-L380">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="JHistint.save_adjacency_matrix-Tuple{Matrix{Float32}, AbstractString}" href="#JHistint.save_adjacency_matrix-Tuple{Matrix{Float32}, AbstractString}"><code>JHistint.save_adjacency_matrix</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">save_adjacency_matrix(matrix::Matrix{Float32}, filepath_matrix::AbstractString)</code></pre><p>Function to save an adjacency matrix represented as a float matrix to a text file.</p><p><strong>Arguments:</strong></p><ul><li><code>matrix::Matrix{Float32}</code>: The float matrix representing the adjacency matrix.</li><li><code>filepath_matrix::AbstractString</code>: The file path represented as a string</li></ul><p>indicating where to save the matrix.</p><p><strong>Notes</strong></p><p>The function opens the file specified by the <code>filepath_matrix</code> path in write mode and writes the matrix in adjacency matrix format, where each row represents the adjacent nodes of a node. The numbers in the matrix are separated by spaces.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/graphManager.jl#L398-L413">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="JHistint.extract_vertex_position-Tuple{MetaGraph}" href="#JHistint.extract_vertex_position-Tuple{MetaGraph}"><code>JHistint.extract_vertex_position</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">extract_vertex_position(G::MetaGraph)</code></pre><p>Extracts the vertex positions from a MetaGraph <code>G</code>.</p><p><strong>Arguments:</strong></p><ul><li><code>G::MetaGraph</code>: The input MetaGraph containing vertices with</li></ul><p>associated positions.</p><p><strong>Return value:</strong></p><ul><li><code>position_array::Vector{Luxor.Point}</code>: An array containing <code>Luxor.Points</code></li></ul><p>representing the positions of the vertices in <code>G</code>.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/graphManager.jl#L431-L443">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="JHistint.extract_vertex_color-Tuple{MetaGraph}" href="#JHistint.extract_vertex_color-Tuple{MetaGraph}"><code>JHistint.extract_vertex_color</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">extract_vertex_color(G::MetaGraph)</code></pre><p>Extracts the vertex colors from a MetaGraph <code>G</code>.</p><p><strong>Arguments:</strong></p><ul><li><code>G::MetaGraph</code>: The input MetaGraph containing vertices with associated colors.</li></ul><p><strong>Return value:</strong></p><ul><li><code>color_array::Vector{Any}</code>: An array containing the extracted color</li></ul><p>information associated with the vertices in <code>G</code>.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/graphManager.jl#L457-L468">source</a></section></article><h2 id="Support-Functions-for-Tessellation-(tessellationManager.jl)"><a class="docs-heading-anchor" href="#Support-Functions-for-Tessellation-(tessellationManager.jl)">Support Functions for Tessellation (tessellationManager.jl)</a><a id="Support-Functions-for-Tessellation-(tessellationManager.jl)-1"></a><a class="docs-heading-anchor-permalink" href="#Support-Functions-for-Tessellation-(tessellationManager.jl)" title="Permalink"></a></h2><article class="docstring"><header><a class="docstring-binding" id="JHistint.build_df_label-Tuple{SegmentedImage, Float32, Float32}" href="#JHistint.build_df_label-Tuple{SegmentedImage, Float32, Float32}"><code>JHistint.build_df_label</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">build_df_label(s::SegmentedImage, min_threshold::Float32, max_threshold::Float32)</code></pre><p>Builds DataFrames containing information about regions in a segmented image, including labels, positions, colors, and areas, and separates them into noisy, filtered, and total regions based on pixel count.</p><p><strong>Arguments :</strong></p><ul><li><code>s::SegmentedImage</code>: The segmented image containing regions.</li><li><code>min_threshold</code>: Minimal threshold for considering segments area.</li><li><code>max_threshold</code>: Maximal threshold for considering segments area.</li></ul><p><strong>Return value :</strong></p><ul><li><code>df_label::DataFrame</code>: A DataFrame containing information about filtered regions</li></ul><p>, regions associated with cell or nuclei.</p><ul><li><code>df_noisy_label::DataFrame</code>: A DataFrame containing information about noisy regions.</li><li><code>df_total_label::DataFrame</code>: A DataFrame containing information about all regions.</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/tessellationManager.jl#L13-L31">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="JHistint.add_column_is_cell-Tuple{DataFrame, DataFrame, DataFrame}" href="#JHistint.add_column_is_cell-Tuple{DataFrame, DataFrame, DataFrame}"><code>JHistint.add_column_is_cell</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">add_column_is_cell(df_labels::DataFrame, df_noisy_labels::DataFrame, df_total_labels::DataFrame)</code></pre><p>Adds a boolean column &#39;is_cell&#39; to the total labels DataFrame, indicating whether each region is a cell or not.</p><p><strong>Arguments:</strong></p><ul><li><code>df_labels::DataFrame</code>: A DataFrame containing information about filtered</li></ul><p>regions (default : areas &gt; 3000 pixels).</p><ul><li><code>df_noisy_labels::DataFrame</code>: A DataFrame containing information about</li></ul><p>noisy regions (default : areas between 300 and 3000 pixels).</p><ul><li><code>df_total_labels::DataFrame</code>: A DataFrame containing information about</li></ul><p>all regions (default : areas &gt; 300 pixels).</p><p><strong>Return value:</strong></p><ul><li><code>df_total_labels::DataFrame</code>: The <code>df_total_labels</code> DataFrame with</li></ul><p>the additional <code>is_cell</code> column.</p><p><strong>Notes:</strong></p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/tessellationManager.jl#L105-L124">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="JHistint.build_dataframe_edges_from_grid-Tuple{Vector{Any}, DataFrame}" href="#JHistint.build_dataframe_edges_from_grid-Tuple{Vector{Any}, DataFrame}"><code>JHistint.build_dataframe_edges_from_grid</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">build_dataframe_edges_from_grid(edge_list::Vector{Any}, df_total_labels::DataFrame)</code></pre><p>Builds a DataFrame containing edge information from a given list of grid-based edges and a DataFrame of total region labels.</p><p><strong>Arguments:</strong></p><ul><li><code>edge_list::Vector{Any}</code>: A list of grid-based edges represented as pairs</li></ul><p>of indices.</p><ul><li><code>df_total_labels::DataFrame</code>: A DataFrame containing information about all</li></ul><p>regions, including labels and areas.</p><p><strong>Return value:</strong></p><ul><li><code>df::DataFrame</code>: A DataFrame containing information about edges,</li></ul><p>including origin, destination, and edge weight.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/tessellationManager.jl#L142-L157">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="JHistint.build_graph_from_tessellation-Tuple{DataFrame, DataFrame, DataFrame, Int64, Int64, AbstractString, AbstractString}" href="#JHistint.build_graph_from_tessellation-Tuple{DataFrame, DataFrame, DataFrame, Int64, Int64, AbstractString, AbstractString}"><code>JHistint.build_graph_from_tessellation</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">build_graph_from_tessellation(df_labels::DataFrame,
                           df_noisy_labels::DataFrame,
                           df_total_labels::DataFrame,
                           w::Int64, h::Int64,
                           filepath_total_tess::AbstractString,
                           filepath_cell_tess::AbstractString)</code></pre><p>Builds a graph rapresentation of the segmented image based on Voronoi tessellations and saves visualizations.</p><p><strong>Arguments:</strong></p><ul><li><code>df_labels::DataFrame</code>: A DataFrame containing information about</li></ul><p>nuclei and centroids.</p><ul><li><code>df_noisy_labels::DataFrame</code>: A DataFrame containing information</li></ul><p>about noisy nuclei centroids.</p><ul><li><code>df_total_labels::DataFrame</code>: A DataFrame containing information</li></ul><p>about all centroids.</p><ul><li><code>w::Int64</code>: Width of the tessellation region.</li><li><code>h::Int64</code>: Height of the tessellation region.</li><li><code>filepath_total_tess::AbstractString</code>: The file path to save the</li></ul><p>visualization of the total tessellation.</p><ul><li><code>filepath_cell_tess::AbstractString</code>: The file path to save the</li></ul><p>visualization of the cell-based tessellation.</p><p><strong>Return value:</strong></p><ul><li><code>df_edges::DataFrame</code>: A DataFrame containing edge information between regions.</li><li><code>edges::Vector{Any}</code>: A vector of pairs representing the connected</li></ul><p>regions based on Voronoi edges.</p><p><strong>Notes:</strong></p><p>The <code>build_graph_from_tessellation</code> function performs the following steps:</p><ol><li>Extracts centroid positions from the provided DataFrames.</li><li>Performs Voronoi tessellations for both the total and cell-based centroids.</li><li>Plots the tessellations, including centroids and labels.</li><li>Saves the visualizations to the specified file paths.</li><li>Constructs an edge DataFrame based on Voronoi edges.</li></ol></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/tessellationManager.jl#L196-L233">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="JHistint.tess_dataframe_to_adjacency_matrix_weight-Tuple{DataFrame, DataFrame, Vector{Any}}" href="#JHistint.tess_dataframe_to_adjacency_matrix_weight-Tuple{DataFrame, DataFrame, Vector{Any}}"><code>JHistint.tess_dataframe_to_adjacency_matrix_weight</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">tess_dataframe_to_adjacency_matrix_weight(df_total_labels::DataFrame, df_edges::DataFrame, edges::Vector{Any})</code></pre><p>Converts a DataFrame representation of edges and region labels into an adjacency matrix with weighted edges.</p><p><strong>Arguments:</strong></p><ul><li><code>df_total_labels::DataFrame</code>: A DataFrame containing information</li></ul><p>about all regions, including labels.</p><ul><li><code>df_edges::DataFrame</code>: A DataFrame containing edge information,</li></ul><p>including origin, destination, and edge weight.</p><ul><li><code>edges::Vector{Any}</code>: A vector of pairs representing the connected regions.</li></ul><p><strong>Return value:</strong></p><ul><li><code>adjacency_matrix::Matrix{Int}</code>: An adjacency matrix representing the</li></ul><p>connectivity of regions with weighted edges.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/tessellationManager.jl#L300-L316">source</a></section></article><h2 id="Support-Functions-for-Noise-Reduction-(noiseManager.jl)"><a class="docs-heading-anchor" href="#Support-Functions-for-Noise-Reduction-(noiseManager.jl)">Support Functions for Noise Reduction (noiseManager.jl)</a><a id="Support-Functions-for-Noise-Reduction-(noiseManager.jl)-1"></a><a class="docs-heading-anchor-permalink" href="#Support-Functions-for-Noise-Reduction-(noiseManager.jl)" title="Permalink"></a></h2><article class="docstring"><header><a class="docstring-binding" id="JHistint.compute_centroid_total_cells-Tuple{SegmentedImage, DataFrame, Float32}" href="#JHistint.compute_centroid_total_cells-Tuple{SegmentedImage, DataFrame, Float32}"><code>JHistint.compute_centroid_total_cells</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">compute_centroid_total_cells(s::SegmentedImage,
                                    df_label::DataFrame,
                                    min_threshold::Float32)</code></pre><p>Computes the centroids of total cells within regions in a segmented image <code>s</code> and associates them with labels in the provided DataFrame <code>df_label</code>.</p><p><strong>Arguments:</strong></p><ul><li><code>s::SegmentedImage</code>: The segmented image containing regions.</li><li><code>df_label::DataFrame</code>: A DataFrame with information about the regions,</li></ul><p>including labels and other attributes.</p><ul><li><code>min_threshold</code>: Minimal threshold for considering segments area.</li></ul><p><strong>Return value:</strong></p><ul><li><code>df_label::DataFrame</code>: The input DataFrame <code>df_label</code> with an additional</li></ul><p><code>centroid</code> column containing the computed centroids of total cells.</p><p><strong>Notes:</strong></p><p>The <code>compute_centroid_total_cells</code> function iterates through the pixels in the segmented image <code>s</code> to identify total cells within regions. It calculates the centroids of these total cells and associates them with their corresponding labels in the <code>df_label</code> DataFrame. The function marks pixels as visited to avoid redundant calculations and applies a manual threshold to exclude noise by considering only regions with pixel counts greater than the specified threshold (default = 300 pixels).</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/noiseManager.jl#L13-L39">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="JHistint.compute_centroid_cells-Tuple{SegmentedImage, DataFrame, Float32}" href="#JHistint.compute_centroid_cells-Tuple{SegmentedImage, DataFrame, Float32}"><code>JHistint.compute_centroid_cells</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">compute_centroid_cells(s::SegmentedImage,
                            df_label::DataFrame,
                            max_threshold::Float32)</code></pre><p>Computes the centroids of only cells associated to nuclei within regions in a segmented image <code>s</code> and associates them with labels in the provided DataFrame <code>df_label</code>.</p><p><strong>Arguments:</strong></p><ul><li><code>s::SegmentedImage</code>: The segmented image containing regions.</li><li><code>df_label::DataFrame</code>: A DataFrame with information about the regions,</li></ul><p>including labels and other attributes.</p><ul><li><code>max_threshold</code>: Maximal threshold for considering segments area.</li></ul><p><strong>Return value:</strong></p><ul><li><code>df_label::DataFrame</code>: The input DataFrame <code>df_label</code> with an additional</li></ul><p><code>centroid</code> column containing the computed centroids of cells.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/noiseManager.jl#L88-L106">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="JHistint.compute_centroid_noisy_cells-Tuple{SegmentedImage, DataFrame, Float32, Float32}" href="#JHistint.compute_centroid_noisy_cells-Tuple{SegmentedImage, DataFrame, Float32, Float32}"><code>JHistint.compute_centroid_noisy_cells</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">compute_centroid_noisy_cells(s::SegmentedImage,
                                    df_label::DataFrame,
                                    min_threshold::Float32,
                                    max_threshold::Float32)</code></pre><p>Computes the centroids of extra or noisy within regions in a segmented image <code>s</code> and associates them with labels in the provided DataFrame <code>df_label</code>.</p><p><strong>Arguments:</strong></p><ul><li><code>s::SegmentedImage</code>: The segmented image containing regions.</li><li><code>df_label::DataFrame</code>: A DataFrame with information about the regions,</li></ul><p>including labels and other attributes.</p><ul><li><code>min_threshold</code>: Minimal threshold for considering segments area.</li><li><code>max_threshold</code>: Maximal threshold for considering segments area.</li></ul><p><strong>Return value:</strong></p><ul><li><code>df_label::DataFrame</code>: The input DataFrame <code>df_label</code> with an additional</li></ul><p><code>centroid</code> column containing the computed centroids of extra cells.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/noiseManager.jl#L155-L174">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="JHistint.filter_dataframe_cells-Tuple{DataFrame, Float32}" href="#JHistint.filter_dataframe_cells-Tuple{DataFrame, Float32}"><code>JHistint.filter_dataframe_cells</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">filter_dataframe_cells(df_label::DataFrame, max_threshold::Float32)</code></pre><p>Filters a DataFrame containing information about regions to retain only cells with areas greater than a specified threshold.</p><p><strong>Arguments:</strong></p><ul><li><code>df_label::DataFrame</code>: The input DataFrame containing region information.</li><li><code>max_threshold</code>: Maximal threshold for considering segments area.</li></ul><p><strong>Return value:</strong></p><ul><li><code>df_filtered::DataFrame</code>: A filtered DataFrame containing information</li></ul><p>about the retained cells, including labels, positions, colors, areas, and centroids.</p><p><strong>Notes:</strong></p><p>The <code>filter_dataframe_cells</code> function takes a DataFrame <code>df_label</code> as input, which should contain information about regions, including labels, positions, colors, areas, and centroids. It filters this DataFrame to retain only those regions (cells) with areas greater a specified threshold (default = 3000 pixels).</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/noiseManager.jl#L225-L245">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="JHistint.filter_dataframe_extras-Tuple{DataFrame, Float32, Float32}" href="#JHistint.filter_dataframe_extras-Tuple{DataFrame, Float32, Float32}"><code>JHistint.filter_dataframe_extras</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">filter_dataframe_extras(df_label::DataFrame,
                                min_threshold::Float32,
                                max_threshold::Float32)</code></pre><p>Filters a DataFrame containing information about regions to retain only extra elements (not cells) with areas between two specified thresholds. In the default case these are 300 and 3000 pixels.</p><p><strong>Arguments:</strong></p><ul><li><code>df_label::DataFrame</code>: The input DataFrame containing region information.</li><li><code>min_threshold</code>: Minimal threshold for considering segments area.</li><li><code>max_threshold</code>: Maximal threshold for considering segments area.</li></ul><p><strong>Return value:</strong></p><ul><li><code>df_filtered::DataFrame</code>: A filtered DataFrame containing information</li></ul><p>about the retained extra elements, including labels, positions, colors, areas, and centroids.</p><p><strong>Notes:</strong></p><p>The <code>filter_dataframe_extras</code> function takes a DataFrame <code>df_label</code> as input, which should contain information about regions, including labels, positions, colors, areas, and centroids. It filters this DataFrame to retain only those regions that are considered &quot;extras&quot; (not cells) and have areas between two specified thresholds.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/noiseManager.jl#L275-L300">source</a></section></article><h2 id="Support-Functions-for-DataBase-(dbManager.jl)"><a class="docs-heading-anchor" href="#Support-Functions-for-DataBase-(dbManager.jl)">Support Functions for DataBase (dbManager.jl)</a><a id="Support-Functions-for-DataBase-(dbManager.jl)-1"></a><a class="docs-heading-anchor-permalink" href="#Support-Functions-for-DataBase-(dbManager.jl)" title="Permalink"></a></h2><article class="docstring"><header><a class="docstring-binding" id="JHistint.insert_record_DB-NTuple{8, AbstractString}" href="#JHistint.insert_record_DB-NTuple{8, AbstractString}"><code>JHistint.insert_record_DB</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">insert_record_DB(col_name::AbstractString,
                      cas_name::AbstractString,
                      tcga_case_id::AbstractString,
                      sin_cas_name::AbstractString,
                      tcga_slide_id::AbstractString,
                      link_slide::AbstractString,
                      filepath_zip::AbstractString,
                      filepath_svs::AbstractString)</code></pre><p>Function for storing in the <code>JHistint_DB</code> database the information associated with each slide downloaded from the Cancer Digital Slide Archive (CDSA).</p><p><strong>Argomenti</strong></p><ul><li><code>col_name::AbstractString</code> = Collection name.</li><li><code>cas_name::AbstractString</code> = Case name, which corresponds to the <code>CASE-NAME</code></li></ul><p>displayed by the package.</p><ul><li><code>tcga_case_id::AbstractString</code> = ID used by TCGA to identify the case.</li><li><code>sin_cas_name::AbstractString</code> = Name of the individual slide, which</li></ul><p>corresponds to the <code>SLIDE-ID</code> displayed by the package.</p><ul><li><code>tcga_slide_id::AbstractString</code> = ID used by TCGA to identify the slide.</li><li><code>link_slide::AbstractString</code> = Link to the APIs for downloading the slide.</li><li><code>filepath_zip::AbstractString</code> = Path where the <code>.zip</code> file is stored.</li><li><code>filepath_svs::AbstractString</code> = Path where the <code>.tif</code> file is stored.</li></ul><p><strong>Notes</strong></p><p>The <code>JHistint_DB</code> database is used for storing the information associated with each slide downloaded from the CDSA. The function takes a dictionary containing the information associated with the slide and stores it in the database. Data available in the <code>JHistint_DB</code> database for each slide:</p><ul><li><code>collection_name TEXT</code> = Name of the collection.</li><li><code>case_name TEXT</code> = Name of the case.</li><li><code>TCGA_caseID TEXT</code> = ID used by TCGA to identify the case.</li><li><code>slide_ID TEXT</code> = Name of the individual slide case.</li><li><code>TCGA_slideID TEXT UNIQUE</code> = ID used by TCGA to identify the slide, <code>UNIQUE</code></li></ul><p>prevents duplicates from being generated.</p><ul><li><code>slide_path_folder_zip TEXT</code> = Path where the <code>.zip</code> file is stored.</li><li><code>slide_path_folder_svs TEXT</code> = Path where the <code>.tif</code> file is stored.</li><li><code>slide_path_api TEXT</code> = Link to the API for downloading the slide.</li><li><code>slide_path_folder_seg TEXT</code> = Path where the segmented <code>.tif</code> file is stored.</li><li><code>slide_svs BLOB</code> = Histopathological slide (image).</li><li><code>slide_info_TSS TEXT</code> = Slide information - Tissue Source Site.</li><li><code>slide_info_participant_code TEXT</code> = Slide information - Participant Code,</li></ul><p>alphanumeric string.</p><ul><li><code>slide_info_sample_type TEXT</code> = Slide information - Sample Type. The values</li></ul><p>associated with tumor samples are in the range 01-09. 10-19 indicates the range for non-diseased normal samples. 20-29 indicates samples currently under control.</p><ul><li><code>slide_info_vial TEXT</code> = Slide information - Vial. Related to the ordering of</li></ul><p>the sample in the sequence of samples. The values range from A-Z.</p><ul><li><code>slide_info_portion TEXT</code> = Slide information - Portion. Related to the</li></ul><p>ordering of the analyzed portions associated with a sample. Takes values in the range 01-99.</p><ul><li><code>slide_info_type TEXT</code> = Slide information - Image Type. The possible</li></ul><p>values are TS (Top Slide), BS (Bottom Slide), and MS (Middle Slide). The alphanumeric value indicates the ordering of the slide.</p><ul><li><code>slide_path_folder_matrix TEXT</code> = Path where the adjacency matrix <code>.txt</code></li></ul><p>file is stored.</p><ul><li><code>matrix_data BLOB</code> = Adjacency matrix.</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/dbManager.jl#L12-L71">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="JHistint.query_extract_slide_svs-Tuple{AbstractString}" href="#JHistint.query_extract_slide_svs-Tuple{AbstractString}"><code>JHistint.query_extract_slide_svs</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">query_extract_slide_svs(collection_name::AbstractString)</code></pre><p>The function queries the <code>JHistint_DB</code> and extracts the list of slides associated with the collection name provided as an argument.</p><p><strong>Arguments</strong></p><ul><li><code>collection_name::AbstractString</code>: Name of the slide collection to search</li></ul><p>for in the <code>JHistint_DB</code>.</p><p><strong>Return value</strong></p><ul><li><code>slide_list</code>: List of tuples, each of which contains the ID of the slide,</li></ul><p>the <code>.svs</code> file of the slide, and the path of the folder containing the <code>.svs</code> file.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/dbManager.jl#L308-L322">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="JHistint.load_seg_slide-Tuple{AbstractString, AbstractString, Matrix{Int64}, AbstractString}" href="#JHistint.load_seg_slide-Tuple{AbstractString, AbstractString, Matrix{Int64}, AbstractString}"><code>JHistint.load_seg_slide</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">load_seg_slide(filepath_seg::AbstractString, filepath_matrix::AbstractString, matrix::Matrix{Int64}, slide_id::AbstractString)</code></pre><p>The function updates the <code>JHistint_DB</code> with the path of the segmented image file, the path of the adjacency matrix file in text format, and the matrix itself.</p><p><strong>Arguments</strong></p><ul><li><code>filepath_seg::AbstractString</code>: Path of the segmented</li></ul><p>image file to add to the DB.</p><ul><li><code>filepath_matrix::AbstractString</code>: Path of the adjacency matrix file.</li><li><code>matrix::Matrix{Int64}</code>: Adjacency matrix.</li><li><code>slide_id::AbstractString</code>: ID of the slide to update with</li></ul><p>the segmented image information.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/dbManager.jl#L338-L351">source</a></section></article><h2 id="SOPHYSM-Support-Functions-for-DataBase-(dbManager.jl)"><a class="docs-heading-anchor" href="#SOPHYSM-Support-Functions-for-DataBase-(dbManager.jl)">SOPHYSM Support Functions for DataBase (dbManager.jl)</a><a id="SOPHYSM-Support-Functions-for-DataBase-(dbManager.jl)-1"></a><a class="docs-heading-anchor-permalink" href="#SOPHYSM-Support-Functions-for-DataBase-(dbManager.jl)" title="Permalink"></a></h2><article class="docstring"><header><a class="docstring-binding" id="JHistint.insert_record_DB_SOPHYSM-NTuple{9, AbstractString}" href="#JHistint.insert_record_DB_SOPHYSM-NTuple{9, AbstractString}"><code>JHistint.insert_record_DB_SOPHYSM</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">insert_record_DB_SOPHYSM(col_name::AbstractString,
                      cas_name::AbstractString,
                      tcga_case_id::AbstractString,
                      sin_cas_name::AbstractString,
                      tcga_slide_id::AbstractString,
                      link_slide::AbstractString,
                      filepath_zip::AbstractString,
                      filepath_svs::AbstractString,
                      path_download_db::AbstractString)</code></pre><p>Function for storing in the <code>JHistint_DB</code> database the information associated with each slide downloaded from the Cancer Digital Slide Archive (CDSA). The function is different from the standard model. It stores only informations of the downloaded Slides.</p><p><strong>Argomenti</strong></p><ul><li><code>col_name::AbstractString</code> = Collection name.</li><li><code>cas_name::AbstractString</code> = Case name, which corresponds to the <code>CASE-NAME</code></li></ul><p>displayed by the package.</p><ul><li><code>tcga_case_id::AbstractString</code> = ID used by TCGA to identify the case.</li><li><code>sin_cas_name::AbstractString</code> = Name of the individual slide, which</li></ul><p>corresponds to the <code>SLIDE-ID</code> displayed by the package.</p><ul><li><code>tcga_slide_id::AbstractString</code> = ID used by TCGA to identify the slide.</li><li><code>link_slide::AbstractString</code> = Link to the APIs for downloading the slide.</li><li><code>filepath_zip::AbstractString</code> = Path where the <code>.zip</code> file is stored.</li><li><code>filepath_svs::AbstractString</code> = Path where the <code>.tif</code> file is stored.</li><li><code>path_download_db::AbstractString</code> = Path where the DB file is stored.</li></ul><p><strong>Notes</strong></p><p>The <code>JHistint_DB</code> database is used for storing the information associated with each slide downloaded from the CDSA. The function takes a dictionary containing the information associated with     the slide and stores it in the database. Data available in the <code>JHistint_DB</code> database for each slide:</p><ul><li><code>collection_name TEXT</code> = Name of the collection.</li><li><code>case_name TEXT</code> = Name of the case.</li><li><code>TCGA_caseID TEXT</code> = ID used by TCGA to identify the case.</li><li><code>slide_ID TEXT</code> = Name of the individual slide case.</li><li><code>TCGA_slideID TEXT UNIQUE</code> = ID used by TCGA to identify the slide,</li></ul><p><code>UNIQUE</code> prevents duplicates from being generated.</p><ul><li><code>slide_path_folder_zip TEXT</code> = Path where the <code>.zip</code> file is stored.</li><li><code>slide_path_folder_svs TEXT</code> = Path where the <code>.tif</code> file is stored.</li><li><code>slide_path_api TEXT</code> = Link to the API for downloading the slide.</li><li><code>slide_path_folder_seg TEXT</code> = Path where the segmented <code>.tif</code> file is stored.</li><li><code>slide_svs BLOB</code> = Histopathological slide (image).</li><li><code>slide_info_TSS TEXT</code> = Slide information - Tissue Source Site.</li><li><code>slide_info_participant_code TEXT</code> = Slide information - Participant Code,</li></ul><p>alphanumeric string.</p><ul><li><code>slide_info_sample_type TEXT</code> = Slide information - Sample Type.</li></ul><p>The values associated with tumor samples are in the range 01-09. 10-19 indicates the range for non-diseased normal samples. 20-29 indicates samples     currently under control.</p><ul><li><code>slide_info_vial TEXT</code> = Slide information - Vial. Related to the ordering</li></ul><p>of the sample in the sequence of samples. The values range from A-Z.</p><ul><li><code>slide_info_portion TEXT</code> = Slide information - Portion. Related to the</li></ul><p>ordering of the analyzed portions associated with a sample. Takes values in the range 01-99.</p><ul><li><code>slide_info_type TEXT</code> = Slide information - Image Type. The possible</li></ul><p>values are TS (Top Slide), BS (Bottom Slide), and MS (Middle Slide). The alphanumeric value indicates the ordering of the slide.</p><ul><li><code>slide_path_folder_matrix TEXT</code> = Path where the adjacency matrix</li></ul><p><code>.txt</code> file is stored.</p><ul><li><code>matrix_data BLOB</code> = Adjacency matrix.</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/dbManager.jl#L165-L229">source</a></section></article><h2 id="API-Support-Functions-(apiManager.jl)"><a class="docs-heading-anchor" href="#API-Support-Functions-(apiManager.jl)">API Support Functions (apiManager.jl)</a><a id="API-Support-Functions-(apiManager.jl)-1"></a><a class="docs-heading-anchor-permalink" href="#API-Support-Functions-(apiManager.jl)" title="Permalink"></a></h2><article class="docstring"><header><a class="docstring-binding" id="JHistint.download_collection_values-Tuple{AbstractString}" href="#JHistint.download_collection_values-Tuple{AbstractString}"><code>JHistint.download_collection_values</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">download_collection_values(filepath::AbstractString)</code></pre><p>Function for downloading data from collections available in TCGA.</p><p><strong>Arguments</strong></p><ul><li><code>filepath::AbstractString</code> = Path where to save the obtained <code>.json</code> file</li></ul><p>from the API available in CDSA.</p><p><strong>Notes</strong></p><p>The API requires the definition of <code>parentType</code> and <code>parentId</code>. <code>parentId</code> specifies the identifier of the collection. The collection of images associated with TCGA is identified by the code: <code>5b9ef8e3e62914002e454c39</code>. The use of <code>limit=0</code> sets the absence of limits in the queried file, ensuring the complete download of the file. The API belongs to the category for managing the folders stored in the repository. The downloaded file is <code>.json</code>.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/apiManager.jl#L14-L31">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="JHistint.extract_collection_values-Tuple{AbstractString}" href="#JHistint.extract_collection_values-Tuple{AbstractString}"><code>JHistint.extract_collection_values</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">extract_collection_values(filepath::AbstractString)</code></pre><p>Function to extract the values of data collections from the <code>.json</code> file downloaded by the <code>download_collection_values</code> function.</p><p><strong>Arguments</strong></p><ul><li><code>filepath::AbstractString</code> = Path where the <code>collectionlist.json</code></li></ul><p>file is stored.</p><p><strong>Return value</strong></p><ul><li><code>collection_values::Array{String}</code> = List of data collections</li></ul><p>available in TCGA.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/apiManager.jl#L47-L60">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="JHistint.download_project_infos-Tuple{AbstractString, AbstractString}" href="#JHistint.download_project_infos-Tuple{AbstractString, AbstractString}"><code>JHistint.download_project_infos</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">download_project_infos(filepath::AbstractString, collection_name::AbstractString)</code></pre><p>Function to download metadata associated with the selected collection at startup.</p><p><strong>Arguments</strong></p><ul><li><code>filepath::AbstractString</code> = Path to save the <code>.json</code> file associated with</li></ul><p>the collection. The file is indicated with the wording <code>collection_name.json</code>.</p><ul><li><code>collection_name::AbstractString</code> = Name of the collection from which to</li></ul><p>download the slides.</p><p><strong>Notes</strong></p><p>The API requires the definition of <code>parentType</code>, <code>parentId</code>, and <code>name</code>. The <code>name</code> attribute identifies the name of the collection from which you want to retrieve data (e.g., chol, esca, etc.). The API belongs to the category for managing the folders stored in the repository. The downloaded file is <code>.json</code>.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/apiManager.jl#L97-L113">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="JHistint.extract_project_id-Tuple{AbstractString}" href="#JHistint.extract_project_id-Tuple{AbstractString}"><code>JHistint.extract_project_id</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">extract_project_id(filepath::AbstractString)</code></pre><p>Function to extract the <code>id</code> value from the metadata of the collection selected at startup.</p><p><strong>Arguments</strong></p><ul><li><code>filepath::AbstractString</code> = Path where the <code>collection_name.json</code> file</li></ul><p>is stored.</p><p><strong>Return value</strong></p><ul><li><code>project_id</code> = <code>id</code> of the collection.</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/apiManager.jl#L127-L139">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="JHistint.getCasesForProject-Tuple{AbstractString, AbstractString}" href="#JHistint.getCasesForProject-Tuple{AbstractString, AbstractString}"><code>JHistint.getCasesForProject</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">getCasesForProject(filepath_case::AbstractString, project_id::AbstractString)</code></pre><p>Function to download metadata associated with the cases of the selected collection at startup.</p><p><strong>Arguments</strong></p><ul><li><code>filepath::AbstractString</code> = Path where to save the <code>.json</code> file</li></ul><p>associated with the cases of the collection. The file is indicated with the term <code>collection_name.json</code>.</p><ul><li><code>project_id::AbstractString</code> = <code>id</code> of the collection.</li></ul><p><strong>Return values</strong></p><ul><li><code>casesID_values</code> = List of <code>id</code> of all the cases in the collection.</li><li><code>casesNAME_values</code> = List of <code>name</code> of all the cases in the collection.</li></ul><p><strong>Notes</strong></p><p>The API requires the definition of <code>parentType</code> and <code>parentId</code>. The <code>parentType</code> attribute is set to <code>folder</code> given the structure of the repository. The <code>parentId</code> is set by defining the identifier of the chosen collection. The downloaded file is <code>.json</code>.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/apiManager.jl#L156-L177">source</a></section></article><article class="docstring"><header><a class="docstring-binding" id="JHistint.download_zip-Tuple{AbstractString, AbstractString}" href="#JHistint.download_zip-Tuple{AbstractString, AbstractString}"><code>JHistint.download_zip</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">download_zip(link::AbstractString, filepath::AbstractString)</code></pre><p>Function for downloading histological slides in <code>.zip</code> format associated with the cases of the selected collection at startup.</p><p><strong>Arguments</strong></p><ul><li><code>link::AbstractString</code> = URL to access the API for slide download.</li><li><code>filepath::AbstractString</code> = Path to save the <code>.zip</code> file.</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/apiManager.jl#L207-L216">source</a></section></article><h2 id="ZIP-Support-Functions-(zipManager.jl)"><a class="docs-heading-anchor" href="#ZIP-Support-Functions-(zipManager.jl)">ZIP Support Functions (zipManager.jl)</a><a id="ZIP-Support-Functions-(zipManager.jl)-1"></a><a class="docs-heading-anchor-permalink" href="#ZIP-Support-Functions-(zipManager.jl)" title="Permalink"></a></h2><article class="docstring"><header><a class="docstring-binding" id="JHistint.extract_slide-Tuple{AbstractString}" href="#JHistint.extract_slide-Tuple{AbstractString}"><code>JHistint.extract_slide</code></a> — <span class="docstring-category">Method</span></header><section><div><pre><code class="language-julia hljs">extract_slide(filepath_zip::AbstractString)</code></pre><p>Function to extract the contents of <code>.zip</code> files downloaded from CDSA.</p><p><strong>Arguments</strong></p><ul><li><code>filepath_zip::AbstractString</code> = Path where the <code>.zip</code> file for</li></ul><p>the individual case is saved.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/niccolo99mandelli/JHistint.jl/blob/67ed9237b0f4ed700db2e7f2a7ccb56887ae090d/src/zipManager.jl#L9-L17">source</a></section></article></article><nav class="docs-footer"><p class="footer-message">Powered by <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> and the <a href="https://julialang.org/">Julia Programming Language</a>.</p></nav></div><div class="modal" id="documenter-settings"><div class="modal-background"></div><div class="modal-card"><header class="modal-card-head"><p class="modal-card-title">Settings</p><button class="delete"></button></header><section class="modal-card-body"><p><label class="label">Theme</label><div class="select"><select id="documenter-themepicker"><option value="documenter-light">documenter-light</option><option value="documenter-dark">documenter-dark</option></select></div></p><hr/><p>This document was generated with <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> version 0.27.24 on <span class="colophon-date" title="Monday 23 October 2023 23:08">Monday 23 October 2023</span>. Using Julia version 1.8.5.</p></section><footer class="modal-card-foot"></footer></div></div></div></body></html>
