var documenterSearchIndex = {"docs":
[{"location":"#JHistint.jl-Julia-Histopathology-Interface","page":"Home","title":"JHistint.jl - Julia Histopathology Interface","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"Julia interface for implementing the REST APIs available on the Cancer Slide Digital Archive (CDSA) portal for downloading histological images available in The Cancer Genome Atlas (TCGA). The Cancer Slide Digital Archive (CDSA) is a web platform for support, sharing, and analysis of digital pathological data. Currently, it hosts over 23,000 images associated with the data available on \"The Cancer Genome Atlas\" Data Portal. The library includes functions for managing image-processing algorithms for cellular and nuclei segmentation, constructing graph and the corresponding adjacency matrix, building tessellation and interfacing with J-Space.jl package to simulate the spatial growth and the genomic evolution of a cell population and the experiment of sequencing the genome of the sampled cells.","category":"page"},{"location":"","page":"Home","title":"Home","text":"Link GitHub repository: JHistint.jl","category":"page"},{"location":"","page":"Home","title":"Home","text":"Link GitHub repository, avaiable on spatial-input branch: J-Space.jl","category":"page"},{"location":"","page":"Home","title":"Home","text":"CDSA Portal: Click Here    ","category":"page"},{"location":"","page":"Home","title":"Home","text":"Repository containing the data mapped in the portal: Click Here","category":"page"},{"location":"","page":"Home","title":"Home","text":"Guide to using the APIs: Click Here","category":"page"},{"location":"#Package-Structure","page":"Home","title":"Package Structure","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"The case and collection folders store metadata in .json format for individual cases and collections available on the TCGA Data Portal. The collection folder is structured as follows:\ncollectionlist.json = Stores access data (metadata) for collections (Projects in TCGA).\ncollection_name.json = Stores access data (metadata) for a single collection. The .json file is generated based on the collection chosen by the user.\nThe case folder is structured as follows:\ncollection_name.json = Stores all metadata related to cases associated with the collection selected by the user.\nThe slides folder stores histological images related to individual cases. The images are organized based on collection (TCGA-chol, TCGA-esca, etc.), and the individual case being analyzed (TCGA-2H-A9GF, TCGA-2H-A9GG, etc.). Within each folder related to the case, the slides are stored in compressed .zip files. The format of each individual slide is .tif. The folder names related to the cases correspond to the values of the Case ID field listed in the TCGA Data Portal. The names of the .zip files located in each folder refer to the Sample ID attribute associated with the patient. The slide name is given by concatenating the Slide ID and Slide UUID attributes that can be found in the lower section of the web page dedicated to the generic case TCGA-XX-YYYY.","category":"page"},{"location":"","page":"Home","title":"Home","text":"Example: TCGA-02-0001-01C-01-TS1.zip  \r\n  - 02 = refers to the TSS (Tissue Source Site).  \r\n  - 0001 = refers to the code associated with the Participant, an alphanumeric string.  \r\n  - 01 = refers to the Sample Type. The values associated with tumor samples are in the range 01-09. 10-19 indicates the range for non-diseased normal samples. 20-29 indicates samples currently under control.  \r\n  - C = refers to the Vial field related to the ordering of the sample in the sample sequence. Values range from A-Z.  \r\n  - 01 = refers to the Portion field related to the ordering of the analyzed portions associated with a sample. It takes values in the range 01-99.  \r\n  - TS1 = refers to the Slide field related to the type of image. The values that can be assumed are TS (Top Slide), BS (Bottom Slide), and MS (Middle Slide). The alphanumeric value indicates the slide ordering.","category":"page"},{"location":"#JHistint-Collections","page":"Home","title":"JHistint Collections","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"The available collections are:","category":"page"},{"location":"","page":"Home","title":"Home","text":"TCGA-BRCA = Breast Invasive Carcinoma (Breast)\nTCGA-OV = Ovarian Serous Cystadenocarcinoma (Ovary)\nTCGA-LUAD = Lung Adenocarcinoma (Bronchus and Lung)\nTCGA-UCEC = Uterine Corpus Endometrial Carcinoma (Corpus uteri)\nTCGA-GBM = Glioblastoma Multiforme (Brain)\nTCGA-HSNC = Head and Neck Squamous Cell Carcinoma (Larynx, Lip, Tonsil, Gum, Other and unspecified parths of mouth)\nTCGA-KIRC = Kidney Renal Clear Cell Carcinoma (Kidney)\nTCGA-LGG = Brain Lower Grade Glioma (Brain)\nTCGA-LUSC = Lung Squamous Cell Carcinoma (Bronchus and lung)\nTCGA-TCHA = Thyroid Carcinoma (Thyroid gland)\nTCGA-PRAD = Prostate Adenocarcinoma (Prostate gland)\nTCGA-SKCM = Skin Cutaneous Melanoma (Skin)\nTCGA-COAD = Colon Adenocarcinoma (Colon)\nTCGA-STAD = Stomach Adenocarcinoma (Stomach)\nTCGA-BLCA = Bladder Urothelial Carcinoma (Bladder)\nTCGA-LIHC = Liver Hepatocellular Carcinoma (Liver and intrahepatic bile ducts)  \nTCGA-CESC = Cervical Squamous Cell Carcinoma and Endocervical Adenocarcinoma (Cervix uteri)\nTCGA-KIRP = Kidney Renal Papillary Cell Carcinoma (Kidney)\nTCGA-SARC = Sarcoma (Various)\nTCGA-ESCA = Esophageal Carcinoma (Esophagus)\nTCGA-PAAD = Pancreatic Adenocarcinoma (Pancreas)\nTCGA-READ = Rectum Adenocarcinoma (Rectum)\nTCGA-PCPG = Pheochromocytoma and Paraganglioma (Adrenal gland)\nTCGA-TGCT = Testicular Germ Cell Tumors (Testis)\nTCGA-THYM = Thymoma (Thymus)\nTCGA-ACC = Adrenocortical Carcinoma -Adenomas and Adenocarcinomas (Adrenal gland)\nTCGA-MESO = Mesothelioma (Heart, mediastinum and pleura)\nTCGA-UVM = Uveal Melanoma (Eye and adnexa)\nTCGA-KICH = Kidney Chromophobe (Kidney)\nTCGA-UCS = Uterine Carcinosarcoma (Uterus, NOS)\nTCGA-CHOL = Cholangiocarcinoma (Liver and intrahepatic bile ducts, Other and unspecified part of biliary track)\nTCGA-DLBC = Lymphoid Neoplasm Diffuse Large B-cell Lymphoma (Various)","category":"page"},{"location":"","page":"Home","title":"Home","text":"To download a specific collection, just indicate the name of the collection: BRCA, OV, LUAD.","category":"page"},{"location":"#Package-Installation","page":"Home","title":"Package Installation","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"Step 1 - Install J-Space from spatial-input branch:","category":"page"},{"location":"","page":"Home","title":"Home","text":"(@v1.8) pkg > add https://github.com/BIMIB-DISCo/J-Space.jl.git#spatial-input","category":"page"},{"location":"","page":"Home","title":"Home","text":"Step 2 - Install JHistint from GitHub Repository:","category":"page"},{"location":"","page":"Home","title":"Home","text":"(@v1.8) pkg > add https://github.com/niccolo99mandelli/JHistint.jl.git\r\njulia > using JHistint","category":"page"},{"location":"#Package-Installation-Julia-Registries-(In-Progress)","page":"Home","title":"Package Installation Julia Registries (In Progress)","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"The JHistint package is available in the Julia Registries and can be installed as follows:","category":"page"},{"location":"","page":"Home","title":"Home","text":"julia > using Pkg\r\njulia > Pkg.add(\"JHistint\")\r\njulia > using JHistint","category":"page"},{"location":"","page":"Home","title":"Home","text":"Otherwise, type ] in the Julia REPL and execute:","category":"page"},{"location":"","page":"Home","title":"Home","text":"(@v1.8) pkg > add JHistint\r\njulia > using JHistint","category":"page"},{"location":"#Download-Slides-Main-Functions-(JHistint.jl)","page":"Home","title":"Download Slides Main Functions (JHistint.jl)","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"download_single_collection(collection_name::AbstractString)","category":"page"},{"location":"#JHistint.download_single_collection-Tuple{AbstractString}","page":"Home","title":"JHistint.download_single_collection","text":"download_single_collection(collection_name::AbstractString)\n\nFunction for downloading histological slides associated with a collection available in TCGA.\n\nArguments\n\ncollection_name::AbstractString = Collection of TCGA data to download\n\nthe histological slides.\n\nNotes\n\nThe function evaluates the collection_name argument, and in case of an invalid collection, considers the configuration in the Config.toml file. The value set in the package is default.\n\n# Examples with valid input\njulia> JHistint.download_single_collection(\"acc\")\njulia> JHistint.download_single_collection(\"bLca\")\n\n# Examples with invalid input\njulia> JHistint.download_single_collection(\"ac\")\njulia> JHistint.download_single_collection(\"\")\n\n\n\n\n\n","category":"method"},{"location":"","page":"Home","title":"Home","text":"download_all_collection()","category":"page"},{"location":"#JHistint.download_all_collection-Tuple{}","page":"Home","title":"JHistint.download_all_collection","text":"download_all_collection()\n\nFunction for downloading histological slides associated with all collections available in TCGA.\n\n# Examples with valid input\njulia> JHistint.download_all_collection()\n\n\n\n\n\n","category":"method"},{"location":"#Download-Slides-SOPHYSM-Functions-(JHistint.jl)","page":"Home","title":"Download Slides SOPHYSM Functions (JHistint.jl)","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"download_single_collection_SOPHYSM(collection_name::AbstractString, path_to_save::AbstractString)","category":"page"},{"location":"#JHistint.download_single_collection_SOPHYSM-Tuple{AbstractString, AbstractString}","page":"Home","title":"JHistint.download_single_collection_SOPHYSM","text":"download_single_collection_SOPHYSM(collection_name::AbstractString, path_to_save::AbstractString)\n\nFunction for downloading histological slides in SOPYHSM_app associated with a collection available in TCGA.\n\nArguments\n\ncollection_name::AbstractString = Collection of TCGA data to download the\n\nhistological slides.\n\npath_to_save::AbstractString = Local folder path for saving\n\nhistological slides.\n\nNotes\n\nThe function evaluates the collection_name argument, and in case of an invalid collection, considers the configuration in the Config.toml file. The value set in the package is default.\n\n# Examples with valid input\njulia> JHistint.download_single_collection_SOPHYSM(\"acc\", \"C:\\...\")\njulia> JHistint.download_single_collection_SOPHYSM(\"bLca\", \"C:\\...\")\n\n# Examples with invalid input\njulia> JHistint.download_single_collection_SOPHYSM(\"ac\", \"C:\\...\")\njulia> JHistint.download_single_collection_SOPHYSM(\"\", \"C:\\...\")\n\n\n\n\n\n","category":"method"},{"location":"","page":"Home","title":"Home","text":"download_all_collection_SOPHYSM(path_to_save::AbstractString)","category":"page"},{"location":"#JHistint.download_all_collection_SOPHYSM-Tuple{AbstractString}","page":"Home","title":"JHistint.download_all_collection_SOPHYSM","text":"download_all_collection_SOPHYSM(path_to_save::AbstractString)\n\nFunction for downloading histological slides associated with all collections available in TCGA.\n\nArguments\n\npath_to_save::AbstractString = Local folder path for saving\n\nhistological slides.\n\n# Examples with valid input\njulia> JHistint.download_all_collection_SOPHYSM(\"C:\\...\")\n\n\n\n\n\n","category":"method"},{"location":"#Cell-Segmentation-Slides-Main-Functions-(JHistint.jl)","page":"Home","title":"Cell Segmentation Slides Main Functions (JHistint.jl)","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"slide_cell_segmentation_without_download(collection_name::AbstractString)","category":"page"},{"location":"#JHistint.slide_cell_segmentation_without_download-Tuple{AbstractString}","page":"Home","title":"JHistint.slide_cell_segmentation_without_download","text":"slide_cell_segmentation_without_download(collection_name::AbstractString)\n\nFunction for performing cell segmentation on histopathological slides present in the JHistint_DB database associated with the collection name provided as an argument. After generating the segmented slide, the function proceeds with constructing and saving the corresponding graph and adjacency matrix.\n\nArguments\n\ncollection_name::AbstractString = Collection of TCGA data to download\n\nthe histological slides.\n\nNotes\n\nThe function utilizes the JHistint_DB database for performing cell segmentation on the histopathological slides associated with the provided collection name. It generates a segmented slide and constructs a corresponding graph and adjacency matrix. The output files are saved in a user-defined directory. The function may take a considerable amount of time to complete, depending on the size of the slides and the complexity of the segmentation algorithm. For each slide in the database, cell segmentation is performed using the apply_segmentation_without_download function, and the path where the result is saved is stored in the database using the load_seg_slide function. The segmentation process is defined in 4 steps:\n\nLOAD SLIDE ... (slide_id)\nAPPLY SEGMENTATION ... (slide_id)\nBUILD GRAPH ... (slide_id)\nBUILD & SAVE ADJACENCY MATRIX ... (slide_id)\nJ-SPACE features ... (slide_id)\n\nThe adjacency matrix is saved in the same directory as the original image in text format. Finally, a confirmation message is printed for each segmented slide. Unlike the slide_cell_segmentation_with_download function, this function does not involve the creation and download of the segmented image.\n\n# Examples with valid input\njulia> JHistint.slide_cell_segmentation_without_download(\"acc\")\njulia> JHistint.slide_cell_segmentation_without_download(\"bLca\")\n\n# Examples with invalid input\njulia> JHistint.slide_cell_segmentation_without_download(\"ac\")\njulia> JHistint.slide_cell_segmentation_without_download(\"\")\n\n\n\n\n\n","category":"method"},{"location":"","page":"Home","title":"Home","text":"slide_cell_segmentation_with_download(collection_name::AbstractString)","category":"page"},{"location":"#JHistint.slide_cell_segmentation_with_download-Tuple{AbstractString}","page":"Home","title":"JHistint.slide_cell_segmentation_with_download","text":"slide_cell_segmentation_with_download(collection_name::AbstractString)\n\nFunction for performing cell segmentation on histopathological slides present in the JHistint_DB database associated with the collection name provided as an argument. The function downloads the segmented slide, which is placed in the same directory as the original slide. After generating the segmented slide, the function proceeds with constructing and saving the corresponding graph and adjacency matrix.\n\nArguments\n\ncollection_name::AbstractString = TCGA data collection for which to\n\nperform cell segmentation.\n\nNotes\n\nThe function utilizes the JHistint_DB database for performing cell segmentation on the histopathological slides associated with the provided collection name. It generates a segmented slide and constructs a corresponding graph and adjacency matrix. The output files are saved in a user-defined directory. The function may take a considerable amount of time to complete, depending on the size of the slides and the complexity of the segmentation algorithm. For each slide in the database, cell segmentation is performed using the apply_segmentation_with_download function, and the path where the result is saved is stored in the database using the load_seg_slide function. The segmentation process is similar to that described in the slide_cell_segmentation_without_download function, with the added step of downloading the segmented image and placing it in the same directory as the original slide. The segmentation process is defined in 6 steps:\n\nLOAD SLIDE ... (slide_id)\nAPPLY SEGMENTATION ... (slide_id)\nBUILD SEGMENTED SLIDE ... (slide_id)\nBUILD GRAPH ... (slide_id)\nBUILD & SAVE ADJACENCY MATRIX ... (slide_id)\nSAVE SEGMENTED SLIDE ... (slide_id)\nJ-SPACE features ... (slide_id)\n\nThe adjacency matrix is saved in text format in the same directory as both the original and segmented images. Finally, a confirmation message is printed for each segmented slide.\n\n# Examples with valid input\njulia> JHistint.slide_cell_segmentation_with_download(\"acc\")\njulia> JHistint.slide_cell_segmentation_with_download(\"bLca\")\n\n# Examples with invalid input\njulia> JHistint.slide_cell_segmentation_with_download(\"ac\")\njulia> JHistint.slide_cell_segmentation_with_download(\"\")\n\n\n\n\n\n","category":"method"},{"location":"#SOPHYSM-Main-Function-(JHistint.jl)","page":"Home","title":"SOPHYSM Main Function (JHistint.jl)","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"start_segmentation_SOPHYSM_tessellation(filepath_input::AbstractString,\r\n                                  filepath_output::AbstractString,\r\n                                  thresholdGray::Float64,\r\n                                  thresholdMarker::Float64,\r\n                                  min_threshold::Float32,\r\n                                  max_threshold::Float32)","category":"page"},{"location":"#JHistint.start_segmentation_SOPHYSM_tessellation-Tuple{AbstractString, AbstractString, Float64, Float64, Float32, Float32}","page":"Home","title":"JHistint.start_segmentation_SOPHYSM_tessellation","text":"start_segmentation_SOPHYSM_tessellation(filepath_input::AbstractString,\n                                filepath_output::AbstractString,\n                                thresholdGray::Float64,\n                                thresholdMarker::Float64,\n                                min_threshold::Float32,\n                                max_threshold::Float32)\n\nInitiates the SOPHYSM segmentation process for histological image using tessellation process.\n\nArguments\n\nfilepath_input::AbstractString: The file path to the input histological\n\nimage to be segmented.\n\nfilepath_output::AbstractString: The file path where the segmented results\n\nand related data will be saved.\n\nthresholdGray::Float64: The grayscale threshold used for initial\n\nimage processing.\n\nthresholdMarker::Float64: The marker threshold for identifying\n\ncellular structures.\n\nmin_threshold: Minimal threshold for considering segments area.\nmax_threshold: Maximal threshold for considering segments area.\n\n\n\n\n\n","category":"method"},{"location":"","page":"Home","title":"Home","text":"start_segmentation_SOPHYSM_graph(filepath_input::AbstractString,\r\n                                  filepath_output::AbstractString,\r\n                                  thresholdGray::Float64,\r\n                                  thresholdMarker::Float64,\r\n                                  min_threshold::Float32,\r\n                                  max_threshold::Float32)","category":"page"},{"location":"#JHistint.start_segmentation_SOPHYSM_graph-Tuple{AbstractString, AbstractString, Float64, Float64, Float32, Float32}","page":"Home","title":"JHistint.start_segmentation_SOPHYSM_graph","text":"start_segmentation_SOPHYSM_graph(filepath_input::AbstractString,\n                                filepath_output::AbstractString,\n                                thresholdGray::Float64,\n                                thresholdMarker::Float64,\n                                min_threshold::Float32,\n                                max_threshold::Float32)\n\nInitiates the SOPHYSM segmentation process for histological image using graph construction.\n\nArguments\n\nfilepath_input::AbstractString: The file path to the input histological\n\nimage to be segmented.\n\nfilepath_output::AbstractString: The file path where the segmented results\n\nand related data will be saved.\n\nthresholdGray::Float64: The grayscale threshold used for initial\n\nimage processing.\n\nthresholdMarker::Float64: The marker threshold for identifying\n\ncellular structures.\n\nmin_threshold: Minimal threshold for considering segments area.\nmax_threshold: Maximal threshold for considering segments area.\n\n\n\n\n\n","category":"method"},{"location":"#Support-Functions-for-Cell-Segmentation-(segmentationManager.jl)","page":"Home","title":"Support Functions for Cell Segmentation (segmentationManager.jl)","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"apply_segmentation_without_download(slide_info::Tuple{String, Vector{UInt8}, String})","category":"page"},{"location":"#JHistint.apply_segmentation_without_download-Tuple{Tuple{String, Vector{UInt8}, String}}","page":"Home","title":"JHistint.apply_segmentation_without_download","text":"apply_segmentation_without_download(slide_info::Tuple{String, Vector{UInt8}, String})\n\nThe function performs the segmentation of a histological image, generates its corresponding graph, and translates it into a symmetric adjacency matrix with only 0s and 1s. Define, also, the dataframe associated with labels and edges.\n\nArguments\n\nslide_info::Tuple{String, Vector{UInt8}, String}: A tuple containing the\n\nslide ID, the image obtained from the DB, and the path of the original image file.\n\nReturn values\n\nfilepath_matrix: The path where the adjacency matrix is stored in .txt format.\nmatrix: The adjacency matrix constructed from the segmentation.\n\nNotes\n\nThe function uses the watershed segmentation algorithm to segment the image into different groups of pixels. Segmentation is performed using a feature transformation of the image (feature_transform) and labeling of connected components. The distance between the different regions is then calculated, and an adjacency graph of the regions is constructed using the region_adjacency_graph function. The resulting graph is then transformed into an Int adjacency matrix using the weighted_graph_to_adjacency_matrix function and saved to the path of the original image.\n\n\n\n\n\n","category":"method"},{"location":"","page":"Home","title":"Home","text":"apply_segmentation_with_download(slide_info::Tuple{String, Vector{UInt8}, String})","category":"page"},{"location":"#JHistint.apply_segmentation_with_download-Tuple{Tuple{String, Vector{UInt8}, String}}","page":"Home","title":"JHistint.apply_segmentation_with_download","text":"apply_segmentation_with_download(slide_info::Tuple{String, Vector{UInt8}, String})\n\nThe function performs segmentation of a histological image, saves the segmented image in .png format, generates the corresponding graph, and translates it into an adjacency matrix. Define, also, the dataframe associated with labels and edges.\n\nArguments\n\nslide_info::Tuple{String, Vector{UInt8}, String}: Tuple containing the\n\nslide ID, the image itself obtained from the DB, and the path of the original image file.\n\nReturn values\n\nfilepath_seg: The path where the segmented image is stored in .tif format.\nfilepath_matrix: The path where the graph is stored in .txt format.\nmatrix: The adjacency matrix constructed from the segmentation.\n\nNotes\n\nThe function uses the watershed segmentation algorithm to segment the image into different groups of pixels. Segmentation is performed using an image feature transformation (feature_transform) and connected component labeling. The distance between different regions is then calculated, and an adjacency graph of the regions is constructed using the region_adjacency_graph function. The obtained graph is transformed into an adjacency matrix using the weighted_graph_to_adjacency_matrix function, which is saved in the path of the original image. Finally, a segmented .png image is saved, and the path of the segmented slide file is returned. It performs also the visualizations of the graph with vertices and edges saved as \"graphvertex.png\" and \"graphedges.png\" in the output directory.\n\n\n\n\n\n","category":"method"},{"location":"","page":"Home","title":"Home","text":"get_random_color(seed)","category":"page"},{"location":"#JHistint.get_random_color-Tuple{Any}","page":"Home","title":"JHistint.get_random_color","text":"get_random_color(seed)\n\nFunction to return a random 8-bit RGB format color, using a specified seed.\n\nArguments\n\nseed: An integer used to initialize the random number generator.\n\nIf two calls to the function use the same seed, the same color will be generated.\n\nReturn value\n\nThe function returns a random 8-bit RGB format color.\n\n\n\n\n\n","category":"method"},{"location":"#SOPHYSM-Support-Function-for-Cell-Segmentation-(segmentationManager.jl)","page":"Home","title":"SOPHYSM Support Function for Cell Segmentation (segmentationManager.jl)","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"apply_segmentation_SOPHYSM_tessellation(filepath_input::AbstractString,\r\n                                  filepath_output::AbstractString,\r\n                                  thresholdGray::Float64,\r\n                                  thresholdMarker::Float64,\r\n                                  min_threshold::Float32,\r\n                                  max_threshold::Float32)","category":"page"},{"location":"#JHistint.apply_segmentation_SOPHYSM_tessellation-Tuple{AbstractString, AbstractString, Float64, Float64, Float32, Float32}","page":"Home","title":"JHistint.apply_segmentation_SOPHYSM_tessellation","text":"apply_segmentation_SOPHYSM_tessellation(filepath_input::AbstractString,\n                                filepath_output::AbstractString,\n                                thresholdGray::Float64,\n                                thresholdMarker::Float64,\n                                min_threshold::Float32,\n                                max_threshold::Float32)\n\nThe function performs segmentation of a histological slide, saves the segmented image in .png format, generates the corresponding tessellation, and translates it into an adjacency matrix and build the corresponding dataframe.\n\nArguments\n\nfilepath_input::AbstractString: The file path to the input image.\nfilepath_output::AbstractString: The file path where the output files will be saved.\nthresholdGray::Float64: The grayscale threshold for image binarization.\nthresholdMarker::Float64: The threshold for marker-based segmentation.\nmin_threshold: Minimal threshold for considering segments area.\nmax_threshold: Maximal threshold for considering segments area.\n\nNotes\n\nThe function uses the watershed segmentation algorithm to segment the image into different groups of pixels. Segmentation is performed using an image feature transformation (feature_transform) and connected component labeling. The apply_segmentation_SOPHYSM function reads an .tif image, applies the SOPHYSM segmentation algorithm, and generates the following outputs:\n\nSegmented image saved as \"_seg.png\" in the output directory.\nDataframe containing label information saved as \"dataframelabels.csv\"\n\nin the output directory. Also, the dataframe contaning extra information about the segment and label computed by the segmentation algorithm.\n\nAdjacency matrix saved as \".txt\" in the output directory.\nDataframe containing edge information saved as \"dataframeedges.csv\"\n\nin the output directory.\n\nVisualizations of the graph with vertices and edges saved as\n\n\"graphvertex.png\" and \"graphedges.png\" in the output directory.\n\n\n\n\n\n","category":"method"},{"location":"","page":"Home","title":"Home","text":"apply_segmentation_SOPHYSM_graph(filepath_input::AbstractString,\r\n                                  filepath_output::AbstractString,\r\n                                  thresholdGray::Float64,\r\n                                  thresholdMarker::Float64,\r\n                                  min_threshold::Float32,\r\n                                  max_threshold::Float32)","category":"page"},{"location":"#JHistint.apply_segmentation_SOPHYSM_graph-Tuple{AbstractString, AbstractString, Float64, Float64, Float32, Float32}","page":"Home","title":"JHistint.apply_segmentation_SOPHYSM_graph","text":"apply_segmentation_SOPHYSM_graph(filepath_input::AbstractString,\n                                filepath_output::AbstractString,\n                                thresholdGray::Float64,\n                                thresholdMarker::Float64,\n                                min_threshold::Float32,\n                                max_threshold::Float32)\n\nThe function performs segmentation of a histological slide, saves the segmented image in .png format, generates the corresponding graph, and translates it into an adjacency matrix and build the corresponding dataframe.\n\nArguments\n\nfilepath_input::AbstractString: The file path to the input image.\nfilepath_output::AbstractString: The file path where the output files will be saved.\nthresholdGray::Float64: The grayscale threshold for image binarization.\nthresholdMarker::Float64: The threshold for marker-based segmentation.\nmin_threshold: Minimal threshold for considering segments area.\nmax_threshold: Maximal threshold for considering segments area.\n\nNotes\n\nThe function uses the watershed segmentation algorithm to segment the image into different groups of pixels. Segmentation is performed using an image feature transformation (feature_transform) and connected component labeling. The apply_segmentation_SOPHYSM function reads an .tif image, applies the SOPHYSM segmentation algorithm, and generates the following outputs:\n\nSegmented image saved as \"_seg.png\" in the output directory.\nDataframe containing label information saved as \"dataframelabels.csv\"\n\nin the output directory. Also, the dataframe contaning extra information about the segment and label computed by the segmentation algorithm.\n\nAdjacency matrix saved as \".txt\" in the output directory.\nDataframe containing edge information saved as \"dataframeedges.csv\"\n\nin the output directory.\n\nVisualizations of the graph with vertices and edges saved as\n\n\"graphvertex.png\" and \"graphedges.png\" in the output directory.\n\n\n\n\n\n","category":"method"},{"location":"#Support-Functions-for-Graph-(graphManager.jl)","page":"Home","title":"Support Functions for Graph (graphManager.jl)","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"region_adjacency_graph_JHistint(s::SegmentedImage, weight_fn::Function,\r\n                                min_threshold::Float32, max_threshold::Float32)","category":"page"},{"location":"#JHistint.region_adjacency_graph_JHistint-Tuple{SegmentedImage, Function, Float32, Float32}","page":"Home","title":"JHistint.region_adjacency_graph_JHistint","text":"region_adjacency_graph_JHistint(s::SegmentedImage, weight_fn::Function,\n                            min_threshold::Float32, max_threshold::Float32)\n\nConstructs a region adjacency graph (RAG) from the SegmentedImage. It returns the RAGalong with a Dict(label=>vertex) map and a dataframe containing the information about the label. weight_fn is used to assign weights to the edges.\n\nArguments :\n\ns::SegmentedImage: The input segmented image containing regions.\nweight_fn::Function: A function that calculates the weight between\n\ntwo adjacent regions. The function should accept two region labels as arguments and return a numeric value representing the weight.\n\nReturn value :\n\nG::SimpleWeightedGraph: The adjacency graph between regions with weights\n\non the edges.\n\nvert_map::Dict{Int, Int}: A dictionary that maps region labels to nodes\n\nin the graph.\n\ndf_label::DataFrame: A DataFrame containing information about regions,\n\nincluding their identifiers, positions, colors, and areas.\n\nNotes:\n\nweight_fn(label1, label2): Returns a real number corresponding to the weight of the edge between label1 and label2.\n\n\n\n\n\n","category":"method"},{"location":"","page":"Home","title":"Home","text":"weighted_graph_to_adjacency_matrix(G::SimpleWeightedGraph{Int64, Float64}, n::Int64)","category":"page"},{"location":"#JHistint.weighted_graph_to_adjacency_matrix-Tuple{SimpleWeightedGraph{Int64, Float64}, Int64}","page":"Home","title":"JHistint.weighted_graph_to_adjacency_matrix","text":"weighted_graph_to_adjacency_matrix(G::SimpleWeightedGraph{Int64, Float64}, n::Int64)\n\nConverts a weighted graph represented as a SimpleWeightedGraph into an unweighted boolean adjacency matrix.\n\nArguments:\n\nG::SimpleWeightedGraph{Int64, Float64}: Weighted graph represented as a\n\nSimpleWeightedGraph with integer vertex labels and floating-point edge weights.\n\nn::Int64: Number of nodes in the adjacency matrix.\n\nReturn value:\n\nadjacency_matrix: Matrix{Int64} boolean adjacency matrix.\n\nNotes:\n\nThe function returns an n x n adjacency matrix representing the unweighted graph. If nodes i and j are adjacent,the adjacency matrix will contain a value of 1 at position (i,j) and (j,i). Otherwise, the adjacency matrix will contain a value of 0.\n\n\n\n\n\n","category":"method"},{"location":"","page":"Home","title":"Home","text":"weighted_graph_to_adjacency_matrix_weight(G::SimpleWeightedGraph{Int64, Float64}, n::Int64)","category":"page"},{"location":"#JHistint.weighted_graph_to_adjacency_matrix_weight-Tuple{SimpleWeightedGraph{Int64, Float64}, Int64}","page":"Home","title":"JHistint.weighted_graph_to_adjacency_matrix_weight","text":"weighted_graph_to_adjacency_matrix_weight(G::SimpleWeightedGraph{Int64, Float64}, n::Int64)\n\nConverts a weighted graph represented as a SimpleWeightedGraph into an weighted adjacency matrix.\n\nArguments:\n\nG::SimpleWeightedGraph{Int64, Float64}: Weighted graph represented as a\n\nSimpleWeightedGraph with integer vertex labels and floating-point edge weights.\n\nn::Int64: Number of nodes in the adjacency matrix.\n\nReturn value:\n\nadjacency_matrix: Matrix{Float32} boolean adjacency matrix.\n\nNotes:\n\nThe function returns an n x n adjacency matrix representing the weighted graph. If nodes i and j are adjacent,the adjacency matrix will contain a value associated with the edge weight at position (i,j) and (j,i). Otherwise, the adjacency matrix will contain a value of -1.\n\n\n\n\n\n","category":"method"},{"location":"","page":"Home","title":"Home","text":"build_dataframe_as_edgelist(mat::Matrix{Int64}, label_list::Vector{Int64})","category":"page"},{"location":"#JHistint.build_dataframe_as_edgelist-Tuple{Matrix{Int64}, Vector{Int64}}","page":"Home","title":"JHistint.build_dataframe_as_edgelist","text":"build_dataframe_as_edgelist(mat::Matrix{Int64}, label_list::Vector{Int64})\n\nBuilds a DataFrame representing an edge list from an input adjacency matrix mat and a list of labels label_list.\n\nArguments:\n\nmat::Matrix{Int64}: The input adjacency matrix where elements represent\n\nconnections between nodes.\n\nlabel_list::Vector{Int64}: A list of node labels to consider when\n\nconstructing the edge list.\n\nReturn value:\n\ndf::DataFrame: A DataFrame representing the edges in the graph,\n\nwith columns 'origin', 'destination', and 'weight' indicating the source node, target node, and edge weight, respectively.\n\nNotes:\n\nThe build_dataframe_as_edgelist function constructs a DataFrame that represents the edges in a graph based on the adjacency matrix mat. It iterates through the upper triangular part of the matrix and adds edges to the DataFrame for non-zero values while considering only nodes with labels present in label_list.\n\n\n\n\n\n","category":"method"},{"location":"","page":"Home","title":"Home","text":"build_dataframe_as_edgelist(mat::Matrix{Float32}, label_list::Vector{Int64})","category":"page"},{"location":"#JHistint.build_dataframe_as_edgelist-Tuple{Matrix{Float32}, Vector{Int64}}","page":"Home","title":"JHistint.build_dataframe_as_edgelist","text":"build_dataframe_as_edgelist(mat::Matrix{Float32}, label_list::Vector{Int64})\n\nBuilds a DataFrame representing an edge list from an input adjacency matrix mat and a list of labels label_list.\n\nArguments:\n\nmat::Matrix{Float32}: The input adjacency matrix where elements represent\n\nconnections between nodes. In this case, the matrix has Float value.\n\nlabel_list::Vector{Int64}: A list of node labels to consider when\n\nconstructing the edge list.\n\nReturn value:\n\ndf::DataFrame: A DataFrame representing the edges in the graph,\n\nwith columns 'origin', 'destination', and 'weight' indicating the source node, target node, and edge weight, respectively.\n\nNotes:\n\nThe build_dataframe_as_edgelist function constructs a DataFrame that represents the edges in a graph based on the adjacency matrix mat. It iterates through the upper triangular part of the matrix and adds edges to the DataFrame for values different from -1 while considering only nodes with labels present in label_list.\n\n\n\n\n\n","category":"method"},{"location":"","page":"Home","title":"Home","text":"save_adjacency_matrix(matrix::Matrix{Int64}, filepath_matrix::AbstractString)","category":"page"},{"location":"#JHistint.save_adjacency_matrix-Tuple{Matrix{Int64}, AbstractString}","page":"Home","title":"JHistint.save_adjacency_matrix","text":"save_adjacency_matrix(matrix::Matrix{Int64}, filepath_matrix::AbstractString)\n\nFunction to save an adjacency matrix represented as an integer matrix to a text file.\n\nArguments:\n\nmatrix::Matrix{Int64}: The integer matrix representing the adjacency matrix.\nfilepath_matrix::AbstractString: The file path represented as a string\n\nindicating where to save the matrix.\n\nNotes\n\nThe function opens the file specified by the filepath_matrix path in write mode and writes the matrix in adjacency matrix format, where each row represents the adjacent nodes of a node. The numbers in the matrix are separated by spaces.\n\n\n\n\n\n","category":"method"},{"location":"","page":"Home","title":"Home","text":"save_adjacency_matrix(matrix::Matrix{Float32}, filepath_matrix::AbstractString)","category":"page"},{"location":"#JHistint.save_adjacency_matrix-Tuple{Matrix{Float32}, AbstractString}","page":"Home","title":"JHistint.save_adjacency_matrix","text":"save_adjacency_matrix(matrix::Matrix{Float32}, filepath_matrix::AbstractString)\n\nFunction to save an adjacency matrix represented as a float matrix to a text file.\n\nArguments:\n\nmatrix::Matrix{Float32}: The float matrix representing the adjacency matrix.\nfilepath_matrix::AbstractString: The file path represented as a string\n\nindicating where to save the matrix.\n\nNotes\n\nThe function opens the file specified by the filepath_matrix path in write mode and writes the matrix in adjacency matrix format, where each row represents the adjacent nodes of a node. The numbers in the matrix are separated by spaces.\n\n\n\n\n\n","category":"method"},{"location":"","page":"Home","title":"Home","text":"extract_vertex_position(G::MetaGraph)","category":"page"},{"location":"#JHistint.extract_vertex_position-Tuple{MetaGraph}","page":"Home","title":"JHistint.extract_vertex_position","text":"extract_vertex_position(G::MetaGraph)\n\nExtracts the vertex positions from a MetaGraph G.\n\nArguments:\n\nG::MetaGraph: The input MetaGraph containing vertices with\n\nassociated positions.\n\nReturn value:\n\nposition_array::Vector{Luxor.Point}: An array containing Luxor.Points\n\nrepresenting the positions of the vertices in G.\n\n\n\n\n\n","category":"method"},{"location":"","page":"Home","title":"Home","text":"extract_vertex_color(G::MetaGraph)","category":"page"},{"location":"#JHistint.extract_vertex_color-Tuple{MetaGraph}","page":"Home","title":"JHistint.extract_vertex_color","text":"extract_vertex_color(G::MetaGraph)\n\nExtracts the vertex colors from a MetaGraph G.\n\nArguments:\n\nG::MetaGraph: The input MetaGraph containing vertices with associated colors.\n\nReturn value:\n\ncolor_array::Vector{Any}: An array containing the extracted color\n\ninformation associated with the vertices in G.\n\n\n\n\n\n","category":"method"},{"location":"#Support-Functions-for-Tessellation-(tessellationManager.jl)","page":"Home","title":"Support Functions for Tessellation (tessellationManager.jl)","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"build_df_label(s::SegmentedImage, min_threshold::Float32, max_threshold::Float32)","category":"page"},{"location":"#JHistint.build_df_label-Tuple{SegmentedImage, Float32, Float32}","page":"Home","title":"JHistint.build_df_label","text":"build_df_label(s::SegmentedImage, min_threshold::Float32, max_threshold::Float32)\n\nBuilds DataFrames containing information about regions in a segmented image, including labels, positions, colors, and areas, and separates them into noisy, filtered, and total regions based on pixel count.\n\nArguments :\n\ns::SegmentedImage: The segmented image containing regions.\nmin_threshold: Minimal threshold for considering segments area.\nmax_threshold: Maximal threshold for considering segments area.\n\nReturn value :\n\ndf_label::DataFrame: A DataFrame containing information about filtered regions\n\n, regions associated with cell or nuclei.\n\ndf_noisy_label::DataFrame: A DataFrame containing information about noisy regions.\ndf_total_label::DataFrame: A DataFrame containing information about all regions.\n\n\n\n\n\n","category":"method"},{"location":"","page":"Home","title":"Home","text":"add_column_is_cell(df_labels::DataFrame, df_noisy_labels::DataFrame, df_total_labels::DataFrame)","category":"page"},{"location":"#JHistint.add_column_is_cell-Tuple{DataFrame, DataFrame, DataFrame}","page":"Home","title":"JHistint.add_column_is_cell","text":"add_column_is_cell(df_labels::DataFrame, df_noisy_labels::DataFrame, df_total_labels::DataFrame)\n\nAdds a boolean column 'is_cell' to the total labels DataFrame, indicating whether each region is a cell or not.\n\nArguments:\n\ndf_labels::DataFrame: A DataFrame containing information about filtered\n\nregions (default : areas > 3000 pixels).\n\ndf_noisy_labels::DataFrame: A DataFrame containing information about\n\nnoisy regions (default : areas between 300 and 3000 pixels).\n\ndf_total_labels::DataFrame: A DataFrame containing information about\n\nall regions (default : areas > 300 pixels).\n\nReturn value:\n\ndf_total_labels::DataFrame: The df_total_labels DataFrame with\n\nthe additional is_cell column.\n\nNotes:\n\n\n\n\n\n","category":"method"},{"location":"","page":"Home","title":"Home","text":"build_dataframe_edges_from_grid(edge_list::Vector{Any}, df_total_labels::DataFrame)","category":"page"},{"location":"#JHistint.build_dataframe_edges_from_grid-Tuple{Vector{Any}, DataFrame}","page":"Home","title":"JHistint.build_dataframe_edges_from_grid","text":"build_dataframe_edges_from_grid(edge_list::Vector{Any}, df_total_labels::DataFrame)\n\nBuilds a DataFrame containing edge information from a given list of grid-based edges and a DataFrame of total region labels.\n\nArguments:\n\nedge_list::Vector{Any}: A list of grid-based edges represented as pairs\n\nof indices.\n\ndf_total_labels::DataFrame: A DataFrame containing information about all\n\nregions, including labels and areas.\n\nReturn value:\n\ndf::DataFrame: A DataFrame containing information about edges,\n\nincluding origin, destination, and edge weight.\n\n\n\n\n\n","category":"method"},{"location":"","page":"Home","title":"Home","text":"build_graph_from_tessellation(df_labels::DataFrame,\r\n                             df_noisy_labels::DataFrame,\r\n                             df_total_labels::DataFrame,\r\n                             w::Int64, h::Int64,\r\n                             filepath_total_tess::AbstractString,\r\n                             filepath_cell_tess::AbstractString)","category":"page"},{"location":"#JHistint.build_graph_from_tessellation-Tuple{DataFrame, DataFrame, DataFrame, Int64, Int64, AbstractString, AbstractString}","page":"Home","title":"JHistint.build_graph_from_tessellation","text":"build_graph_from_tessellation(df_labels::DataFrame,\n                           df_noisy_labels::DataFrame,\n                           df_total_labels::DataFrame,\n                           w::Int64, h::Int64,\n                           filepath_total_tess::AbstractString,\n                           filepath_cell_tess::AbstractString)\n\nBuilds a graph rapresentation of the segmented image based on Voronoi tessellations and saves visualizations.\n\nArguments:\n\ndf_labels::DataFrame: A DataFrame containing information about\n\nnuclei and centroids.\n\ndf_noisy_labels::DataFrame: A DataFrame containing information\n\nabout noisy nuclei centroids.\n\ndf_total_labels::DataFrame: A DataFrame containing information\n\nabout all centroids.\n\nw::Int64: Width of the tessellation region.\nh::Int64: Height of the tessellation region.\nfilepath_total_tess::AbstractString: The file path to save the\n\nvisualization of the total tessellation.\n\nfilepath_cell_tess::AbstractString: The file path to save the\n\nvisualization of the cell-based tessellation.\n\nReturn value:\n\ndf_edges::DataFrame: A DataFrame containing edge information between regions.\nedges::Vector{Any}: A vector of pairs representing the connected\n\nregions based on Voronoi edges.\n\nNotes:\n\nThe build_graph_from_tessellation function performs the following steps:\n\nExtracts centroid positions from the provided DataFrames.\nPerforms Voronoi tessellations for both the total and cell-based centroids.\nPlots the tessellations, including centroids and labels.\nSaves the visualizations to the specified file paths.\nConstructs an edge DataFrame based on Voronoi edges.\n\n\n\n\n\n","category":"method"},{"location":"","page":"Home","title":"Home","text":"tess_dataframe_to_adjacency_matrix_weight(df_total_labels::DataFrame, df_edges::DataFrame, edges::Vector{Any})","category":"page"},{"location":"#JHistint.tess_dataframe_to_adjacency_matrix_weight-Tuple{DataFrame, DataFrame, Vector{Any}}","page":"Home","title":"JHistint.tess_dataframe_to_adjacency_matrix_weight","text":"tess_dataframe_to_adjacency_matrix_weight(df_total_labels::DataFrame, df_edges::DataFrame, edges::Vector{Any})\n\nConverts a DataFrame representation of edges and region labels into an adjacency matrix with weighted edges.\n\nArguments:\n\ndf_total_labels::DataFrame: A DataFrame containing information\n\nabout all regions, including labels.\n\ndf_edges::DataFrame: A DataFrame containing edge information,\n\nincluding origin, destination, and edge weight.\n\nedges::Vector{Any}: A vector of pairs representing the connected regions.\n\nReturn value:\n\nadjacency_matrix::Matrix{Int}: An adjacency matrix representing the\n\nconnectivity of regions with weighted edges.\n\n\n\n\n\n","category":"method"},{"location":"#Support-Functions-for-Noise-Reduction-(noiseManager.jl)","page":"Home","title":"Support Functions for Noise Reduction (noiseManager.jl)","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"compute_centroid_total_cells(s::SegmentedImage,\r\n                                      df_label::DataFrame,\r\n                                      min_threshold::Float32)","category":"page"},{"location":"#JHistint.compute_centroid_total_cells-Tuple{SegmentedImage, DataFrame, Float32}","page":"Home","title":"JHistint.compute_centroid_total_cells","text":"compute_centroid_total_cells(s::SegmentedImage,\n                                    df_label::DataFrame,\n                                    min_threshold::Float32)\n\nComputes the centroids of total cells within regions in a segmented image s and associates them with labels in the provided DataFrame df_label.\n\nArguments:\n\ns::SegmentedImage: The segmented image containing regions.\ndf_label::DataFrame: A DataFrame with information about the regions,\n\nincluding labels and other attributes.\n\nmin_threshold: Minimal threshold for considering segments area.\n\nReturn value:\n\ndf_label::DataFrame: The input DataFrame df_label with an additional\n\ncentroid column containing the computed centroids of total cells.\n\nNotes:\n\nThe compute_centroid_total_cells function iterates through the pixels in the segmented image s to identify total cells within regions. It calculates the centroids of these total cells and associates them with their corresponding labels in the df_label DataFrame. The function marks pixels as visited to avoid redundant calculations and applies a manual threshold to exclude noise by considering only regions with pixel counts greater than the specified threshold (default = 300 pixels).\n\n\n\n\n\n","category":"method"},{"location":"","page":"Home","title":"Home","text":"compute_centroid_cells(s::SegmentedImage,\r\n                              df_label::DataFrame,\r\n                              max_threshold::Float32)","category":"page"},{"location":"#JHistint.compute_centroid_cells-Tuple{SegmentedImage, DataFrame, Float32}","page":"Home","title":"JHistint.compute_centroid_cells","text":"compute_centroid_cells(s::SegmentedImage,\n                            df_label::DataFrame,\n                            max_threshold::Float32)\n\nComputes the centroids of only cells associated to nuclei within regions in a segmented image s and associates them with labels in the provided DataFrame df_label.\n\nArguments:\n\ns::SegmentedImage: The segmented image containing regions.\ndf_label::DataFrame: A DataFrame with information about the regions,\n\nincluding labels and other attributes.\n\nmax_threshold: Maximal threshold for considering segments area.\n\nReturn value:\n\ndf_label::DataFrame: The input DataFrame df_label with an additional\n\ncentroid column containing the computed centroids of cells.\n\n\n\n\n\n","category":"method"},{"location":"","page":"Home","title":"Home","text":"compute_centroid_noisy_cells(s::SegmentedImage,\r\n                                      df_label::DataFrame,\r\n                                      min_threshold::Float32,\r\n                                      max_threshold::Float32)","category":"page"},{"location":"#JHistint.compute_centroid_noisy_cells-Tuple{SegmentedImage, DataFrame, Float32, Float32}","page":"Home","title":"JHistint.compute_centroid_noisy_cells","text":"compute_centroid_noisy_cells(s::SegmentedImage,\n                                    df_label::DataFrame,\n                                    min_threshold::Float32,\n                                    max_threshold::Float32)\n\nComputes the centroids of extra or noisy within regions in a segmented image s and associates them with labels in the provided DataFrame df_label.\n\nArguments:\n\ns::SegmentedImage: The segmented image containing regions.\ndf_label::DataFrame: A DataFrame with information about the regions,\n\nincluding labels and other attributes.\n\nmin_threshold: Minimal threshold for considering segments area.\nmax_threshold: Maximal threshold for considering segments area.\n\nReturn value:\n\ndf_label::DataFrame: The input DataFrame df_label with an additional\n\ncentroid column containing the computed centroids of extra cells.\n\n\n\n\n\n","category":"method"},{"location":"","page":"Home","title":"Home","text":"filter_dataframe_cells(df_label::DataFrame, max_threshold::Float32)","category":"page"},{"location":"#JHistint.filter_dataframe_cells-Tuple{DataFrame, Float32}","page":"Home","title":"JHistint.filter_dataframe_cells","text":"filter_dataframe_cells(df_label::DataFrame, max_threshold::Float32)\n\nFilters a DataFrame containing information about regions to retain only cells with areas greater than a specified threshold.\n\nArguments:\n\ndf_label::DataFrame: The input DataFrame containing region information.\nmax_threshold: Maximal threshold for considering segments area.\n\nReturn value:\n\ndf_filtered::DataFrame: A filtered DataFrame containing information\n\nabout the retained cells, including labels, positions, colors, areas, and centroids.\n\nNotes:\n\nThe filter_dataframe_cells function takes a DataFrame df_label as input, which should contain information about regions, including labels, positions, colors, areas, and centroids. It filters this DataFrame to retain only those regions (cells) with areas greater a specified threshold (default = 3000 pixels).\n\n\n\n\n\n","category":"method"},{"location":"","page":"Home","title":"Home","text":"filter_dataframe_extras(df_label::DataFrame,\r\n                                  min_threshold::Float32,\r\n                                  max_threshold::Float32)","category":"page"},{"location":"#JHistint.filter_dataframe_extras-Tuple{DataFrame, Float32, Float32}","page":"Home","title":"JHistint.filter_dataframe_extras","text":"filter_dataframe_extras(df_label::DataFrame,\n                                min_threshold::Float32,\n                                max_threshold::Float32)\n\nFilters a DataFrame containing information about regions to retain only extra elements (not cells) with areas between two specified thresholds. In the default case these are 300 and 3000 pixels.\n\nArguments:\n\ndf_label::DataFrame: The input DataFrame containing region information.\nmin_threshold: Minimal threshold for considering segments area.\nmax_threshold: Maximal threshold for considering segments area.\n\nReturn value:\n\ndf_filtered::DataFrame: A filtered DataFrame containing information\n\nabout the retained extra elements, including labels, positions, colors, areas, and centroids.\n\nNotes:\n\nThe filter_dataframe_extras function takes a DataFrame df_label as input, which should contain information about regions, including labels, positions, colors, areas, and centroids. It filters this DataFrame to retain only those regions that are considered \"extras\" (not cells) and have areas between two specified thresholds.\n\n\n\n\n\n","category":"method"},{"location":"#Support-Functions-for-DataBase-(dbManager.jl)","page":"Home","title":"Support Functions for DataBase (dbManager.jl)","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"insert_record_DB(col_name::AbstractString,\r\n                        cas_name::AbstractString,\r\n                        tcga_case_id::AbstractString,\r\n                        sin_cas_name::AbstractString,\r\n                        tcga_slide_id::AbstractString,\r\n                        link_slide::AbstractString,\r\n                        filepath_zip::AbstractString,\r\n                        filepath_svs::AbstractString)","category":"page"},{"location":"#JHistint.insert_record_DB-NTuple{8, AbstractString}","page":"Home","title":"JHistint.insert_record_DB","text":"insert_record_DB(col_name::AbstractString,\n                      cas_name::AbstractString,\n                      tcga_case_id::AbstractString,\n                      sin_cas_name::AbstractString,\n                      tcga_slide_id::AbstractString,\n                      link_slide::AbstractString,\n                      filepath_zip::AbstractString,\n                      filepath_svs::AbstractString)\n\nFunction for storing in the JHistint_DB database the information associated with each slide downloaded from the Cancer Digital Slide Archive (CDSA).\n\nArgomenti\n\ncol_name::AbstractString = Collection name.\ncas_name::AbstractString = Case name, which corresponds to the CASE-NAME\n\ndisplayed by the package.\n\ntcga_case_id::AbstractString = ID used by TCGA to identify the case.\nsin_cas_name::AbstractString = Name of the individual slide, which\n\ncorresponds to the SLIDE-ID displayed by the package.\n\ntcga_slide_id::AbstractString = ID used by TCGA to identify the slide.\nlink_slide::AbstractString = Link to the APIs for downloading the slide.\nfilepath_zip::AbstractString = Path where the .zip file is stored.\nfilepath_svs::AbstractString = Path where the .tif file is stored.\n\nNotes\n\nThe JHistint_DB database is used for storing the information associated with each slide downloaded from the CDSA. The function takes a dictionary containing the information associated with the slide and stores it in the database. Data available in the JHistint_DB database for each slide:\n\ncollection_name TEXT = Name of the collection.\ncase_name TEXT = Name of the case.\nTCGA_caseID TEXT = ID used by TCGA to identify the case.\nslide_ID TEXT = Name of the individual slide case.\nTCGA_slideID TEXT UNIQUE = ID used by TCGA to identify the slide, UNIQUE\n\nprevents duplicates from being generated.\n\nslide_path_folder_zip TEXT = Path where the .zip file is stored.\nslide_path_folder_svs TEXT = Path where the .tif file is stored.\nslide_path_api TEXT = Link to the API for downloading the slide.\nslide_path_folder_seg TEXT = Path where the segmented .tif file is stored.\nslide_svs BLOB = Histopathological slide (image).\nslide_info_TSS TEXT = Slide information - Tissue Source Site.\nslide_info_participant_code TEXT = Slide information - Participant Code,\n\nalphanumeric string.\n\nslide_info_sample_type TEXT = Slide information - Sample Type. The values\n\nassociated with tumor samples are in the range 01-09. 10-19 indicates the range for non-diseased normal samples. 20-29 indicates samples currently under control.\n\nslide_info_vial TEXT = Slide information - Vial. Related to the ordering of\n\nthe sample in the sequence of samples. The values range from A-Z.\n\nslide_info_portion TEXT = Slide information - Portion. Related to the\n\nordering of the analyzed portions associated with a sample. Takes values in the range 01-99.\n\nslide_info_type TEXT = Slide information - Image Type. The possible\n\nvalues are TS (Top Slide), BS (Bottom Slide), and MS (Middle Slide). The alphanumeric value indicates the ordering of the slide.\n\nslide_path_folder_matrix TEXT = Path where the adjacency matrix .txt\n\nfile is stored.\n\nmatrix_data BLOB = Adjacency matrix.\n\n\n\n\n\n","category":"method"},{"location":"","page":"Home","title":"Home","text":"query_extract_slide_svs(collection_name::AbstractString)","category":"page"},{"location":"#JHistint.query_extract_slide_svs-Tuple{AbstractString}","page":"Home","title":"JHistint.query_extract_slide_svs","text":"query_extract_slide_svs(collection_name::AbstractString)\n\nThe function queries the JHistint_DB and extracts the list of slides associated with the collection name provided as an argument.\n\nArguments\n\ncollection_name::AbstractString: Name of the slide collection to search\n\nfor in the JHistint_DB.\n\nReturn value\n\nslide_list: List of tuples, each of which contains the ID of the slide,\n\nthe .svs file of the slide, and the path of the folder containing the .svs file.\n\n\n\n\n\n","category":"method"},{"location":"","page":"Home","title":"Home","text":"load_seg_slide(filepath_seg::AbstractString, filepath_matrix::AbstractString, matrix::Matrix{Int64}, slide_id::AbstractString)","category":"page"},{"location":"#JHistint.load_seg_slide-Tuple{AbstractString, AbstractString, Matrix{Int64}, AbstractString}","page":"Home","title":"JHistint.load_seg_slide","text":"load_seg_slide(filepath_seg::AbstractString, filepath_matrix::AbstractString, matrix::Matrix{Int64}, slide_id::AbstractString)\n\nThe function updates the JHistint_DB with the path of the segmented image file, the path of the adjacency matrix file in text format, and the matrix itself.\n\nArguments\n\nfilepath_seg::AbstractString: Path of the segmented\n\nimage file to add to the DB.\n\nfilepath_matrix::AbstractString: Path of the adjacency matrix file.\nmatrix::Matrix{Int64}: Adjacency matrix.\nslide_id::AbstractString: ID of the slide to update with\n\nthe segmented image information.\n\n\n\n\n\n","category":"method"},{"location":"#SOPHYSM-Support-Functions-for-DataBase-(dbManager.jl)","page":"Home","title":"SOPHYSM Support Functions for DataBase (dbManager.jl)","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"insert_record_DB_SOPHYSM(col_name::AbstractString,\r\n                        cas_name::AbstractString,\r\n                        tcga_case_id::AbstractString,\r\n                        sin_cas_name::AbstractString,\r\n                        tcga_slide_id::AbstractString,\r\n                        link_slide::AbstractString,\r\n                        filepath_zip::AbstractString,\r\n                        filepath_svs::AbstractString,\r\n                        path_download_db::AbstractString)","category":"page"},{"location":"#JHistint.insert_record_DB_SOPHYSM-NTuple{9, AbstractString}","page":"Home","title":"JHistint.insert_record_DB_SOPHYSM","text":"insert_record_DB_SOPHYSM(col_name::AbstractString,\n                      cas_name::AbstractString,\n                      tcga_case_id::AbstractString,\n                      sin_cas_name::AbstractString,\n                      tcga_slide_id::AbstractString,\n                      link_slide::AbstractString,\n                      filepath_zip::AbstractString,\n                      filepath_svs::AbstractString,\n                      path_download_db::AbstractString)\n\nFunction for storing in the JHistint_DB database the information associated with each slide downloaded from the Cancer Digital Slide Archive (CDSA). The function is different from the standard model. It stores only informations of the downloaded Slides.\n\nArgomenti\n\ncol_name::AbstractString = Collection name.\ncas_name::AbstractString = Case name, which corresponds to the CASE-NAME\n\ndisplayed by the package.\n\ntcga_case_id::AbstractString = ID used by TCGA to identify the case.\nsin_cas_name::AbstractString = Name of the individual slide, which\n\ncorresponds to the SLIDE-ID displayed by the package.\n\ntcga_slide_id::AbstractString = ID used by TCGA to identify the slide.\nlink_slide::AbstractString = Link to the APIs for downloading the slide.\nfilepath_zip::AbstractString = Path where the .zip file is stored.\nfilepath_svs::AbstractString = Path where the .tif file is stored.\npath_download_db::AbstractString = Path where the DB file is stored.\n\nNotes\n\nThe JHistint_DB database is used for storing the information associated with each slide downloaded from the CDSA. The function takes a dictionary containing the information associated with     the slide and stores it in the database. Data available in the JHistint_DB database for each slide:\n\ncollection_name TEXT = Name of the collection.\ncase_name TEXT = Name of the case.\nTCGA_caseID TEXT = ID used by TCGA to identify the case.\nslide_ID TEXT = Name of the individual slide case.\nTCGA_slideID TEXT UNIQUE = ID used by TCGA to identify the slide,\n\nUNIQUE prevents duplicates from being generated.\n\nslide_path_folder_zip TEXT = Path where the .zip file is stored.\nslide_path_folder_svs TEXT = Path where the .tif file is stored.\nslide_path_api TEXT = Link to the API for downloading the slide.\nslide_path_folder_seg TEXT = Path where the segmented .tif file is stored.\nslide_svs BLOB = Histopathological slide (image).\nslide_info_TSS TEXT = Slide information - Tissue Source Site.\nslide_info_participant_code TEXT = Slide information - Participant Code,\n\nalphanumeric string.\n\nslide_info_sample_type TEXT = Slide information - Sample Type.\n\nThe values associated with tumor samples are in the range 01-09. 10-19 indicates the range for non-diseased normal samples. 20-29 indicates samples     currently under control.\n\nslide_info_vial TEXT = Slide information - Vial. Related to the ordering\n\nof the sample in the sequence of samples. The values range from A-Z.\n\nslide_info_portion TEXT = Slide information - Portion. Related to the\n\nordering of the analyzed portions associated with a sample. Takes values in the range 01-99.\n\nslide_info_type TEXT = Slide information - Image Type. The possible\n\nvalues are TS (Top Slide), BS (Bottom Slide), and MS (Middle Slide). The alphanumeric value indicates the ordering of the slide.\n\nslide_path_folder_matrix TEXT = Path where the adjacency matrix\n\n.txt file is stored.\n\nmatrix_data BLOB = Adjacency matrix.\n\n\n\n\n\n","category":"method"},{"location":"#API-Support-Functions-(apiManager.jl)","page":"Home","title":"API Support Functions (apiManager.jl)","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"download_collection_values(filepath::AbstractString)","category":"page"},{"location":"#JHistint.download_collection_values-Tuple{AbstractString}","page":"Home","title":"JHistint.download_collection_values","text":"download_collection_values(filepath::AbstractString)\n\nFunction for downloading data from collections available in TCGA.\n\nArguments\n\nfilepath::AbstractString = Path where to save the obtained .json file\n\nfrom the API available in CDSA.\n\nNotes\n\nThe API requires the definition of parentType and parentId. parentId specifies the identifier of the collection. The collection of images associated with TCGA is identified by the code: 5b9ef8e3e62914002e454c39. The use of limit=0 sets the absence of limits in the queried file, ensuring the complete download of the file. The API belongs to the category for managing the folders stored in the repository. The downloaded file is .json.\n\n\n\n\n\n","category":"method"},{"location":"","page":"Home","title":"Home","text":"extract_collection_values(filepath::AbstractString)","category":"page"},{"location":"#JHistint.extract_collection_values-Tuple{AbstractString}","page":"Home","title":"JHistint.extract_collection_values","text":"extract_collection_values(filepath::AbstractString)\n\nFunction to extract the values of data collections from the .json file downloaded by the download_collection_values function.\n\nArguments\n\nfilepath::AbstractString = Path where the collectionlist.json\n\nfile is stored.\n\nReturn value\n\ncollection_values::Array{String} = List of data collections\n\navailable in TCGA.\n\n\n\n\n\n","category":"method"},{"location":"","page":"Home","title":"Home","text":"download_project_infos(filepath::AbstractString, collection_name::AbstractString)","category":"page"},{"location":"#JHistint.download_project_infos-Tuple{AbstractString, AbstractString}","page":"Home","title":"JHistint.download_project_infos","text":"download_project_infos(filepath::AbstractString, collection_name::AbstractString)\n\nFunction to download metadata associated with the selected collection at startup.\n\nArguments\n\nfilepath::AbstractString = Path to save the .json file associated with\n\nthe collection. The file is indicated with the wording collection_name.json.\n\ncollection_name::AbstractString = Name of the collection from which to\n\ndownload the slides.\n\nNotes\n\nThe API requires the definition of parentType, parentId, and name. The name attribute identifies the name of the collection from which you want to retrieve data (e.g., chol, esca, etc.). The API belongs to the category for managing the folders stored in the repository. The downloaded file is .json.\n\n\n\n\n\n","category":"method"},{"location":"","page":"Home","title":"Home","text":"extract_project_id(filepath::AbstractString)","category":"page"},{"location":"#JHistint.extract_project_id-Tuple{AbstractString}","page":"Home","title":"JHistint.extract_project_id","text":"extract_project_id(filepath::AbstractString)\n\nFunction to extract the id value from the metadata of the collection selected at startup.\n\nArguments\n\nfilepath::AbstractString = Path where the collection_name.json file\n\nis stored.\n\nReturn value\n\nproject_id = id of the collection.\n\n\n\n\n\n","category":"method"},{"location":"","page":"Home","title":"Home","text":"getCasesForProject(filepath_case::AbstractString, project_id::AbstractString)","category":"page"},{"location":"#JHistint.getCasesForProject-Tuple{AbstractString, AbstractString}","page":"Home","title":"JHistint.getCasesForProject","text":"getCasesForProject(filepath_case::AbstractString, project_id::AbstractString)\n\nFunction to download metadata associated with the cases of the selected collection at startup.\n\nArguments\n\nfilepath::AbstractString = Path where to save the .json file\n\nassociated with the cases of the collection. The file is indicated with the term collection_name.json.\n\nproject_id::AbstractString = id of the collection.\n\nReturn values\n\ncasesID_values = List of id of all the cases in the collection.\ncasesNAME_values = List of name of all the cases in the collection.\n\nNotes\n\nThe API requires the definition of parentType and parentId. The parentType attribute is set to folder given the structure of the repository. The parentId is set by defining the identifier of the chosen collection. The downloaded file is .json.\n\n\n\n\n\n","category":"method"},{"location":"","page":"Home","title":"Home","text":"download_zip(link::AbstractString, filepath::AbstractString)","category":"page"},{"location":"#JHistint.download_zip-Tuple{AbstractString, AbstractString}","page":"Home","title":"JHistint.download_zip","text":"download_zip(link::AbstractString, filepath::AbstractString)\n\nFunction for downloading histological slides in .zip format associated with the cases of the selected collection at startup.\n\nArguments\n\nlink::AbstractString = URL to access the API for slide download.\nfilepath::AbstractString = Path to save the .zip file.\n\n\n\n\n\n","category":"method"},{"location":"#ZIP-Support-Functions-(zipManager.jl)","page":"Home","title":"ZIP Support Functions (zipManager.jl)","text":"","category":"section"},{"location":"","page":"Home","title":"Home","text":"extract_slide(filepath_zip::AbstractString)","category":"page"},{"location":"#JHistint.extract_slide-Tuple{AbstractString}","page":"Home","title":"JHistint.extract_slide","text":"extract_slide(filepath_zip::AbstractString)\n\nFunction to extract the contents of .zip files downloaded from CDSA.\n\nArguments\n\nfilepath_zip::AbstractString = Path where the .zip file for\n\nthe individual case is saved.\n\n\n\n\n\n","category":"method"}]
}
